<html><head><META http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"><title>Part 1. Overview of GNA</title><link href="javahelp.css" rel="stylesheet" type="text/css"><meta content="DocBook XSL Stylesheets V1.75.2" name="generator"><link rel="home" href="index.html" title="Genetic Network Analyzer Tutorial"><link rel="up" href="index.html" title="Genetic Network Analyzer Tutorial"><link rel="prev" href="index.html" title="Genetic Network Analyzer Tutorial"><link rel="next" href="ch01s02.html" title="1.2.&nbsp;Projects using GNA"></head><body bgcolor="white" text="black" link="#0000FF" vlink="#840084" alink="#0000FF"><div class="chapter" title="Part 1. Overview of GNA"><div class="titlepage"><div><div><h2 class="title"><a name="iogma.gna.help.overview"></a>Part 1. Overview of GNA</h2></div></div></div><div class="section" title="1.1.&nbsp;What is GNA?"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.overview.whatis_gna"></a>1.1.&nbsp;What is GNA?</h2></div></div></div><p><span><span class="bold"><strong>GNA (Genetic Network
    Analyzer)</strong></span> is a computer tool for the </span>modeling,
    simulation and analysis of genetic regulatory networks<span>. The aim of
    GNA is to assist biologists and bioinformaticians in constructing a model
    of a regulatory network using knowledge about regulatory interactions in
    combination with gene expression data.</span></p><p><span>GNA consists of a simulator of </span><a class="ulink" href="http://ibis.inrialpes.fr/rubrique155.html?menu=menu2" target="_top"><span class="Hyperlink">qualitative models of genetic regulatory
    networks</span></a><span lang=""> in the form of piecewise-linear
    differential equations, a class of models originally introduced by Leon
    Glass and Stuart Kauffman. The simulator has been implemented in Java and
    has been applied to the analysis of various regulatory systems, including
    the networks controlling the initiation of sporulation in <span class="emphasis"><em>B.
    subtilis</em></span> and carbon starvation in <span class="emphasis"><em>E. coli</em></span>.
    The simulator can be used in combination with standard model-checking
    tools in order to analyze important model properties.</span></p><p><span>GNA has been developped at <a class="ulink" href="http://www.inrialpes.fr" target="_top">INRIA Grenoble-Rh&ocirc;ne-Alpes</a>,
    initially by <a class="ulink" href="http://ibis.inrialpes.fr/article165.html" target="_top">Hidde
    de Jong</a> and <a class="ulink" href="http://www-helix.inrialpes.fr/article206.html" target="_top">Michel Page</a>.
    The following persons have contributed to the current and previous
    versions of the software: Gr&eacute;gory Batt (now at INRIA Rocquencourt), Bruno
    Besson (now at Genostar SA), Estelle Dumas, C&eacute;line Hernandez (now at the
    Swiss Institute for Bioinformatics, Geneva) and Pedro Monteiro (now at
    IST, Portugal). <a class="ulink" href="mailto:Jean-Luc.Gouze@sophia.inria.fr" target="_top">Jean-Luc Gouz&eacute;</a> (INRIA
    Sophia Antipolis) and <a class="ulink" href="http://www.math.uha.fr/sari/" target="_top">Tewfik
    Sari</a> (Universit&eacute; de Haute-Alsace, Mulhouse) have made
    contributions to the qualitative simulation method underlying the tool,
    while <a class="ulink" href="http://ibis.inrialpes.fr/article856.html" target="_top">Johannes
    Geiselmann</a> (Universit&eacute; Joseph Fourier, Grenoble) and <a class="ulink" href="http://ibis.inrialpes.fr/article593.html" target="_top">Delphine Ropers</a>
    have worked on applications to actual regulatory networks. GNA has
    recently been transferred to <a class="ulink" href="http://www.genostar.com/" target="_top">Genostar</a>, but remains freely
    available for non profit academic research purposes.</span></p><p><span>General information on the <span class="bold"><strong>qualitative
    modeling and simulation of genetic regulatory networks</strong></span> can be
    found in: </span></p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span lang="">H. de Jong (2002), </span><a class="ulink" href="http://ibis.inrialpes.fr/article341.html" target="_top"><span lang="" class="Hyperlink">Modeling and simulation of genetic regulatory
        systems: A literature review</span></a><span lang="">,
        <span class="emphasis"><em>Journal of Computational Biology</em></span>, 9(1):69-105.
        </span></p></li><li class="listitem" style="list-style-type: disc"><p>H. de Jong, D. Ropers (2005), Qualitative approaches towards the
        analysis of genetic regulatory networks, Z. Szallasi, V. Periwal, J.
        Stelling (eds), <span class="emphasis"><em>System Modeling in Cellular Biology: From
        Concepts to Nuts and Bolts</em></span>, MIT Press, Cambridge, MA.
        125-148.</p></li></ul></div><p>More information on the <span class="bold"><strong>qualitative simulation
    method and its extensions</strong></span> can be found in:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span lang="">H. de Jong, J.-L. Gouz&eacute;, C. Hernandez, M. Page,
        T. Sari, and J. Geiselmann (2004), </span><a class="ulink" href="http://ibis.inrialpes.fr/article532.html" target="_top"><span lang="" class="Hyperlink">Qualitative simulation of genetic regulatory networks
        using piecewise-linear models</span></a><span lang="">,
        <span class="emphasis"><em>Bulletin of Mathematical Biology</em></span>, 66(2):301-340.
        </span></p></li><li class="listitem" style="list-style-type: disc"><p>G. Batt, H. de Jong, M. Page, J. Geiselmann (2008), <a class="ulink" href="http://ibis.inrialpes.fr/article839.html" target="_top">Symbolic reachability
        analysis of genetic regulatory networks using qualitative
        abstractions</a>, <span class="emphasis"><em>Automatica</em></span>.
        44(4):982-989.</p></li><li class="listitem" style="list-style-type: disc"><p>G. Batt, D. Ropers, H. de Jong, J. Geiselmann, R. Mateescu, M.
        Page, D. Schneider (2005), <a class="ulink" href="http://ibis.inrialpes.fr/article727.html" target="_top">Validation of
        qualitative models of genetic regulatory networks by model checking:
        Analysis of the nutritional stress response in <span class="emphasis"><em>Escherichia
        coli</em></span></a>, <span class="emphasis"><em>Bioinformatics</em></span>, 21(Suppl
        1):i19-i28.</p></li><li class="listitem" style="list-style-type: disc"><p>R. Casey, H. de Jong, J.-L. Gouz&eacute; (2006), <a class="ulink" href="http://ibis.inrialpes.fr/article743.html" target="_top">Piecewise-linear models
        of genetic regulatory networks: Equilibria and their
        stability</a>, <span class="emphasis"><em>Journal of Mathematical
        Biology</em></span>. 52(1):124-152.</p></li><li class="listitem" style="list-style-type: disc"><p>H. de Jong, M. Page (2008), <a class="ulink" href="http://ibis.inrialpes.fr/article840.html" target="_top">Search for steady
        states of piecewise-linear differential equation models of genetic
        regulatory networks</a>, <span class="emphasis"><em>ACM/IEEE Transactions on
        Computational Biology and Bioinformatics</em></span>. 5(2):
        208-222.</p></li><li class="listitem" style="list-style-type: disc"><p>P.T. Monteiro, D. Ropers, R. Mateescu, A.T. Freitas, H. de Jong
        (2008), <a class="ulink" href="http://ibis.inrialpes.fr/article913.html" target="_top">Temporal
        logic patterns for querying dynamic models of cellular interaction
        networks</a>, <span class="emphasis"><em>Bioinformatics</em></span>, 24(16)
        :i227-i233.</p></li><li class="listitem" style="list-style-type: disc"><p>R. Mateescu, P.T. Monteiro, E. Dumas, H. de Jong (2010), CTRL:
        Extension of CTL with regular expressions and fairness operators to
        verify genetic regulatory networks, Theoretical Computer Science,
        accepted for publication. Special issue <a class="ulink" href="http://wwwmosi.informatik.uni-rostock.de/cmsb08" target="_top">CMSB
        2008</a>.</p></li><li class="listitem" style="list-style-type: disc"><p>P.T. Monteiro, E. Dumas, B. Besson, R. Mateescu, M. Page, A.T.
        Freitas, H. de Jong (2009), <a class="ulink" href="http://ibis.inrialpes.fr/article952.html" target="_top">A service-oriented
        architecture for integrating the modeling and formal verification of
        genetic regulatory networks</a>, <span class="emphasis"><em>BMC
        Bioinformatics</em></span>, 10:450.</p></li><li class="listitem" style="list-style-type: disc"><p>V. Baldazzi, D. Ropers, Y. Markowicz, D. Kahn, J. Geiselmann, H.
        de Jong (2010), <a class="ulink" href="http://ibis.inrialpes.fr/article973.html" target="_top">The carbon assimilation
        network in Escherichia coli is densely connected and largely
        sign-determined by directions of metabolic fluxes</a>,
        <span class="emphasis"><em>PLoS Computational Biology</em></span>, 6(6) : e1000812.
        </p></li></ul></div><p><span>A previous version of </span><span class="bold"><strong>GNA
    </strong></span>is described in:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span lang="">H. de Jong, J. Geiselmann, C. Hernandez, and M.
        Page (2003), </span><a class="ulink" href="http://www-helix.inrialpes.fr/article348.html" target="_top"><span lang="" class="Hyperlink">Genetic Network Analyzer: Qualitative simulation of
        genetic regulatory networks</span></a><span lang="">,<span class="emphasis"><em> Bioinformatics</em></span>, 19(3):336-344.
        </span></p></li></ul></div><p>The <span class="bold"><strong>application of GNA</strong></span> is
    illustrated in:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span lang="">H. de Jong, J. Geiselmann, G. Batt, C.
        Hernandez, and M. Page (2004), </span><a class="ulink" href="http://ibis.inrialpes.fr/article533.html" target="_top"><span lang="" class="Hyperlink">Qualitative simulation of the initiation of
        sporulation in <span class="emphasis"><em>Bacillus
        subtilis</em></span></span></a><span lang="">,
        <span class="emphasis"><em>Bulletin of Mathematical Biology</em></span>, 66(2):261-300.
        </span></p></li><li class="listitem" style="list-style-type: disc"><p>D. Ropers, H. de Jong, M. Page, D. Schneider, J. Geiselmann
        (2006), <a class="ulink" href="http://ibis.inrialpes.fr/article742.html" target="_top">Qualitative simulation
        of the carbon starvation response in Escherichia coli</a>,
        <span class="emphasis"><em>Biosystems</em></span>, 84(2):124-152.</p></li><li class="listitem" style="list-style-type: disc"><p>J-A. Sepulchre, S. Reverchon, W. Nasser (2007), Modeling the
        onset of virulence in a pectinolytic bacterium, <span class="emphasis"><em>Journal of
        Theoretical Biology</em></span>, 44(2):239-257.</p></li><li class="listitem" style="list-style-type: disc"><p>A. Usseglio Viretta, M. Fussenegger (2004), Modeling the quorum
        sensing regulatory network of human-pathogenic <span class="emphasis"><em>Pseudomonas
        aeruginosa</em></span>, <span class="emphasis"><em>Biotechnology Progress</em></span>,
        20(3):670-678.</p></li></ul></div><p>An introduction to the use of GNA, with modeling tricks, can be
    found in:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p>G. Batt, B. Besson, P. Ciron, H. de Jong, E. Dumas, J.
          Geiselmann, R. Monte, P.T. Monteiro, M. Page, F. Rechenmann, D.
          Ropers, Genetic Network Analyzer: A Tool for the Qualitative
          Modeling and Simulation of Bacterial Regulatory Networks. In D.
          Thieffry et al (eds), <span class="emphasis"><em>Bacterial Molecular Networks:
          Methods and Protocols</em></span>, Human press, to appear.</p></li></ul></div><p>GNA uses the following open-source software: <a class="ulink" href="http://www.jgraph.com" target="_top">JGraph</a>, <a class="ulink" href="http://www.sat4j.org" target="_top">SAT4J</a>, and <a class="ulink" href="http://www2.cs.tum.edu/projects/cup/" target="_top">CUP</a>.The model-checker
    web server at INRIA runs <a class="ulink" href="http://nusmv.fbk.eu/" target="_top">NuSMV</a>.</p></div></div></body></html>