<html><head><META http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"><title>Genetic Network Analyzer Tutorial</title><link href="style.css" rel="stylesheet" type="text/css"><meta content="DocBook XSL Stylesheets V1.75.2" name="generator"></head><body bgcolor="white" text="black" link="#0000FF" vlink="#840084" alink="#0000FF"><div id="banner"></div><div style="float:left;"><a href="http://www.inria.fr"><img src="images/logo_inria.jpg"></a><a href="http://www.genostar.com"><img src="images/logo_genostar2.jpg"></a></div><div style="float:right;">31 July 2012</div><div style="clear:both;"></div><div lang="en" class="book" title="Genetic Network Analyzer Tutorial"><div class="titlepage"><div><div><h1 class="title"><a name="iogma.gna.help"></a>Genetic Network Analyzer Tutorial</h1></div></div></div><div class="toc"><p><b>Table of Contents</b></p><dl><dt><span class="chapter"><a href="#iogma.gna.help.overview">1. Overview of GNA</a></span></dt><dd><dl><dt><span class="section"><a href="#iogma.gna.help.overview.whatis_gna">1.1. What is GNA?</a></span></dt><dt><span class="section"><a href="#iogma.gna.help.overview.projects">1.2. Projects using GNA</a></span></dt></dl></dd><dt><span class="chapter"><a href="#iogma.gna.help.using">2. Using GNA</a></span></dt><dd><dl><dt><span class="section"><a href="#iogma.gna.help.using.defining_network">2.1. Defining a network</a></span></dt><dd><dl><dt><span class="section"><a href="#N1018A">2.1.1. Define simple genetic regulatory network</a></span></dt><dt><span class="section"><a href="#N101CC">2.1.2. Define more complex regulatory networks</a></span></dt><dt><span class="section"><a href="#N101F6">2.1.3. Reduction of regulatory network</a></span></dt></dl></dd><dt><span class="section"><a href="#iogma.gna.help.using.building_model">2.2. Building a model</a></span></dt><dd><dl><dt><span class="section"><a href="#iogman.gna.using.building_model.definition_of_variables">2.2.1. Definition of state and input variables</a></span></dt><dt><span class="section"><a href="#iogman.gna.using.building_model.definition_of_parameters">2.2.2. Definition of parameters</a></span></dt><dt><span class="section"><a href="#iogman.gna.using.building_model.definition_of_equations">2.2.3. Definition of state equations</a></span></dt><dt><span class="section"><a href="#iogma.gna.help.using.parameters_inequalities">2.2.4. Definition of inequality constraints</a></span></dt><dt><span class="section"><a href="#N104C9">2.2.5. Definition of the model through the wizard</a></span></dt></dl></dd><dt><span class="section"><a href="#iogma.gna.help.using.building_initial_conditions">2.3. Specifying initial conditions</a></span></dt><dt><span class="section"><a href="#iogma.gna.help.using.running_simulations">2.4. Running simulations</a></span></dt><dt><span class="section"><a href="#iogma.gna.help.using.interpreting_results">2.5. Interpreting simulation results</a></span></dt><dt><span class="section"><a href="#iogma.gna.help.using.attractors">2.6. Searching for attractors</a></span></dt><dt><span class="section"><a href="#iogma.gna.help.using.building_property">2.7. Checking properties of the state transition graph</a></span></dt><dd><dl><dt><span class="section"><a href="#iogma.gna.help.using.building_atomic_proposition">2.7.1. Specification of atomic propositions</a></span></dt><dt><span class="section"><a href="#iogma.gna.help.using.formulating_temporal_logic_formulas">2.7.2. Formulation of temporal logic formulas</a></span></dt><dt><span class="section"><a href="#iogma.gna.help.using.ready_for_model_checking">2.7.3. Verification of temporal logic formulas</a></span></dt></dl></dd><dt><span class="section"><a href="#iogma.gna.help.using.gna_functions">2.8. Functions of the graphical user interface</a></span></dt><dd><dl><dt><span class="section"><a href="#N109D6">2.8.1. Network editor</a></span></dt><dt><span class="section"><a href="#N10B34">2.8.2. Model editor</a></span></dt></dl></dd></dl></dd><dt><span class="chapter"><a href="#N10D69">3. Appendices</a></span></dt><dd><dl><dt><span class="section"><a href="#iogma.gna.help.appendix.model_syntax">3.1. Syntax of GNA model</a></span></dt><dd><dl><dt><span class="section"><a href="#N10D77">3.1.1. Syntactic conventions</a></span></dt><dt><span class="section"><a href="#N10D9B">3.1.2. BNF of a model</a></span></dt><dt><span class="section"><a href="#N10F34">3.1.3. Additional constraints</a></span></dt></dl></dd><dt><span class="section"><a href="#iogma.gna.help.appendix.initialconditions_syntax">3.2. Syntax of GNA initial conditions</a></span></dt><dd><dl><dt><span class="section"><a href="#N10FED">3.2.1. Syntactic conventions</a></span></dt><dt><span class="section"><a href="#N11011">3.2.2. BNF of an initial conditions file</a></span></dt><dt><span class="section"><a href="#N11090">3.2.3. Additional constraints</a></span></dt></dl></dd><dt><span class="section"><a href="#iogma.gna.help.appendix.property_syntax">3.3. Syntax of CTRL property</a></span></dt><dd><dl><dt><span class="section"><a href="#N110BD">3.3.1. Syntactic conventions</a></span></dt><dt><span class="section"><a href="#N110E1">3.3.2. BNF of a property</a></span></dt><dt><span class="section"><a href="#N11208">3.3.3. Additional constraints</a></span></dt><dt><span class="section"><a href="#N1122B">3.3.4. BNF of a property formula</a></span></dt></dl></dd><dt><span class="section"><a href="#N114D2">3.4. CUP Parser Generator license</a></span></dt><dt><span class="section"><a href="#N114DE">3.5. JGraph license</a></span></dt><dt><span class="section"><a href="#N114F6">3.6. SAT4J license</a></span></dt><dt><span class="section"><a href="#N1150A">3.7. NuSMV license</a></span></dt></dl></dd></dl></div><div class="list-of-figures"><p><b>List of Figures</b></p><dl><dt>2.1. <a href="#figure.network">Schema of simple genetic regulatory network</a></dt><dt>2.2. <a href="#figure.complex_network">Schema of a more complex genetic regulatory
        network</a></dt><dt>2.3. <a href="#figure.reduced_network">Schema of reduced genetic regulatory
        network</a></dt><dt>2.4. <a href="#figure.reduced_network_window"><span class="guilabel">Reduced network</span> window</a></dt><dt>2.5. <a href="#figure.variable_window_param"><span class="guilabel">Variable</span> window after creation of
        the parameters</a></dt><dt>2.6. <a href="#figure.variable_window_equa"><span class="guilabel">Variable</span> window after definition of
        the state equation</a></dt><dt>2.7. <a href="#figure.variable_window_inequal"><span class="guilabel">Variable</span> window after definition of
        the inequality constraints</a></dt><dt>2.8. <a href="#figure.wizard1"><span class="guilabel">Wizard</span> window for the definition of
        variables</a></dt><dt>2.9. <a href="#figure.wizard2"><span class="guilabel">Wizard</span> window for the definition of
        state equation</a></dt><dt>2.10. <a href="#figure.initial_conditions"><span class="guilabel">Initial conditions</span>
      window</a></dt><dt>2.11. <a href="#figure.simulation_window"><span class="guilabel">Simulation</span> window</a></dt><dt>2.12. <a href="#figure.simulation_window_results">Simulation results</a></dt><dt>2.13. <a href="#figure.simulation_window_graph">Simulation results: fragment of the state transition
      graph</a></dt><dt>2.14. <a href="#figure.simulation_window_path">Qualitative evolution of the variables</a></dt><dt>2.15. <a href="#figure.attractor_search"><span class="guilabel">Attractor search</span> window</a></dt><dt>2.16. <a href="#figure.atomic_proposition_editor"><span class="guilabel">Atomic proposition</span>
        windows</a></dt><dt>2.17. <a href="#figure.property_editor_pattern"><span class="guilabel">Pattern</span> wizard of the property
        editor</a></dt><dt>2.18. <a href="#figure.property_editor_expert"><span class="guilabel">Expert</span> tab of the property
        editor</a></dt><dt>2.19. <a href="#figure.go_for_model_checking"><span class="guilabel">Verification</span> window</a></dt><dt>2.20. <a href="#figure.toolbar_network">Menu of the graphical user interface (network
        editor)</a></dt><dt>2.21. <a href="#figure.toolbar_model">Menu of the graphical user interface (model
        editor)</a></dt></dl></div><div class="chapter" title="Part 1. Overview of GNA"><div class="titlepage"><div><div><h2 class="title"><a name="iogma.gna.help.overview"></a>Part 1. Overview of GNA</h2></div></div></div><div class="section" title="1.1.&nbsp;What is GNA?"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.overview.whatis_gna"></a>1.1.&nbsp;What is GNA?</h2></div></div></div><p><span><span class="bold"><strong>GNA (Genetic Network
    Analyzer)</strong></span> is a computer tool for the </span>modeling,
    simulation and analysis of genetic regulatory networks<span>. The aim of
    GNA is to assist biologists and bioinformaticians in constructing a model
    of a regulatory network using knowledge about regulatory interactions in
    combination with gene expression data.</span></p><p><span>GNA consists of a simulator of </span><a class="ulink" href="http://ibis.inrialpes.fr/rubrique155.html?menu=menu2" target="_top"><span class="Hyperlink">qualitative models of genetic regulatory
    networks</span></a><span lang=""> in the form of piecewise-linear
    differential equations, a class of models originally introduced by Leon
    Glass and Stuart Kauffman. The simulator has been implemented in Java and
    has been applied to the analysis of various regulatory systems, including
    the networks controlling the initiation of sporulation in <span class="emphasis"><em>B.
    subtilis</em></span> and carbon starvation in <span class="emphasis"><em>E. coli</em></span>.
    The simulator can be used in combination with standard model-checking
    tools in order to analyze important model properties.</span></p><p><span>GNA has been developped at <a class="ulink" href="http://www.inrialpes.fr" target="_top">INRIA Grenoble-Rh&ocirc;ne-Alpes</a>,
    initially by <a class="ulink" href="http://ibis.inrialpes.fr/article165.html" target="_top">Hidde
    de Jong</a> and <a class="ulink" href="http://www-helix.inrialpes.fr/article206.html" target="_top">Michel Page</a>.
    The following persons have contributed to the current and previous
    versions of the software: Gr&eacute;gory Batt (now at INRIA Rocquencourt), Bruno
    Besson (now at Genostar SA), Estelle Dumas, C&eacute;line Hernandez (now at the
    Swiss Institute for Bioinformatics, Geneva) and Pedro Monteiro (now at
    IST, Portugal). <a class="ulink" href="mailto:Jean-Luc.Gouze@sophia.inria.fr" target="_top">Jean-Luc Gouz&eacute;</a> (INRIA
    Sophia Antipolis) and <a class="ulink" href="http://www.math.uha.fr/sari/" target="_top">Tewfik
    Sari</a> (Universit&eacute; de Haute-Alsace, Mulhouse) have made
    contributions to the qualitative simulation method underlying the tool,
    while <a class="ulink" href="http://ibis.inrialpes.fr/article856.html" target="_top">Johannes
    Geiselmann</a> (Universit&eacute; Joseph Fourier, Grenoble) and <a class="ulink" href="http://ibis.inrialpes.fr/article593.html" target="_top">Delphine Ropers</a>
    have worked on applications to actual regulatory networks. GNA has
    recently been transferred to <a class="ulink" href="http://www.genostar.com/" target="_top">Genostar</a>, but remains freely
    available for non profit academic research purposes.</span></p><p><span>General information on the <span class="bold"><strong>qualitative
    modeling and simulation of genetic regulatory networks</strong></span> can be
    found in: </span></p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span lang="">H. de Jong (2002), </span><a class="ulink" href="http://ibis.inrialpes.fr/article341.html" target="_top"><span lang="" class="Hyperlink">Modeling and simulation of genetic regulatory
        systems: A literature review</span></a><span lang="">,
        <span class="emphasis"><em>Journal of Computational Biology</em></span>, 9(1):69-105.
        </span></p></li><li class="listitem" style="list-style-type: disc"><p>H. de Jong, D. Ropers (2005), Qualitative approaches towards the
        analysis of genetic regulatory networks, Z. Szallasi, V. Periwal, J.
        Stelling (eds), <span class="emphasis"><em>System Modeling in Cellular Biology: From
        Concepts to Nuts and Bolts</em></span>, MIT Press, Cambridge, MA.
        125-148.</p></li></ul></div><p>More information on the <span class="bold"><strong>qualitative simulation
    method and its extensions</strong></span> can be found in:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span lang="">H. de Jong, J.-L. Gouz&eacute;, C. Hernandez, M. Page,
        T. Sari, and J. Geiselmann (2004), </span><a class="ulink" href="http://ibis.inrialpes.fr/article532.html" target="_top"><span lang="" class="Hyperlink">Qualitative simulation of genetic regulatory networks
        using piecewise-linear models</span></a><span lang="">,
        <span class="emphasis"><em>Bulletin of Mathematical Biology</em></span>, 66(2):301-340.
        </span></p></li><li class="listitem" style="list-style-type: disc"><p>G. Batt, H. de Jong, M. Page, J. Geiselmann (2008), <a class="ulink" href="http://ibis.inrialpes.fr/article839.html" target="_top">Symbolic reachability
        analysis of genetic regulatory networks using qualitative
        abstractions</a>, <span class="emphasis"><em>Automatica</em></span>.
        44(4):982-989.</p></li><li class="listitem" style="list-style-type: disc"><p>G. Batt, D. Ropers, H. de Jong, J. Geiselmann, R. Mateescu, M.
        Page, D. Schneider (2005), <a class="ulink" href="http://ibis.inrialpes.fr/article727.html" target="_top">Validation of
        qualitative models of genetic regulatory networks by model checking:
        Analysis of the nutritional stress response in <span class="emphasis"><em>Escherichia
        coli</em></span></a>, <span class="emphasis"><em>Bioinformatics</em></span>, 21(Suppl
        1):i19-i28.</p></li><li class="listitem" style="list-style-type: disc"><p>R. Casey, H. de Jong, J.-L. Gouz&eacute; (2006), <a class="ulink" href="http://ibis.inrialpes.fr/article743.html" target="_top">Piecewise-linear models
        of genetic regulatory networks: Equilibria and their
        stability</a>, <span class="emphasis"><em>Journal of Mathematical
        Biology</em></span>. 52(1):124-152.</p></li><li class="listitem" style="list-style-type: disc"><p>H. de Jong, M. Page (2008), <a class="ulink" href="http://ibis.inrialpes.fr/article840.html" target="_top">Search for steady
        states of piecewise-linear differential equation models of genetic
        regulatory networks</a>, <span class="emphasis"><em>ACM/IEEE Transactions on
        Computational Biology and Bioinformatics</em></span>. 5(2):
        208-222.</p></li><li class="listitem" style="list-style-type: disc"><p>P.T. Monteiro, D. Ropers, R. Mateescu, A.T. Freitas, H. de Jong
        (2008), <a class="ulink" href="http://ibis.inrialpes.fr/article913.html" target="_top">Temporal
        logic patterns for querying dynamic models of cellular interaction
        networks</a>, <span class="emphasis"><em>Bioinformatics</em></span>, 24(16)
        :i227-i233.</p></li><li class="listitem" style="list-style-type: disc"><p>R. Mateescu, P.T. Monteiro, E. Dumas, H. de Jong (2010), CTRL:
        Extension of CTL with regular expressions and fairness operators to
        verify genetic regulatory networks, Theoretical Computer Science,
        accepted for publication. Special issue <a class="ulink" href="http://wwwmosi.informatik.uni-rostock.de/cmsb08" target="_top">CMSB
        2008</a>.</p></li><li class="listitem" style="list-style-type: disc"><p>P.T. Monteiro, E. Dumas, B. Besson, R. Mateescu, M. Page, A.T.
        Freitas, H. de Jong (2009), <a class="ulink" href="http://ibis.inrialpes.fr/article952.html" target="_top">A service-oriented
        architecture for integrating the modeling and formal verification of
        genetic regulatory networks</a>, <span class="emphasis"><em>BMC
        Bioinformatics</em></span>, 10:450.</p></li><li class="listitem" style="list-style-type: disc"><p>V. Baldazzi, D. Ropers, Y. Markowicz, D. Kahn, J. Geiselmann, H.
        de Jong (2010), <a class="ulink" href="http://ibis.inrialpes.fr/article973.html" target="_top">The carbon assimilation
        network in Escherichia coli is densely connected and largely
        sign-determined by directions of metabolic fluxes</a>,
        <span class="emphasis"><em>PLoS Computational Biology</em></span>, 6(6) : e1000812.
        </p></li></ul></div><p><span>A previous version of </span><span class="bold"><strong>GNA
    </strong></span>is described in:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span lang="">H. de Jong, J. Geiselmann, C. Hernandez, and M.
        Page (2003), </span><a class="ulink" href="http://www-helix.inrialpes.fr/article348.html" target="_top"><span lang="" class="Hyperlink">Genetic Network Analyzer: Qualitative simulation of
        genetic regulatory networks</span></a><span lang="">,<span class="emphasis"><em> Bioinformatics</em></span>, 19(3):336-344.
        </span></p></li></ul></div><p>The <span class="bold"><strong>application of GNA</strong></span> is
    illustrated in:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span lang="">H. de Jong, J. Geiselmann, G. Batt, C.
        Hernandez, and M. Page (2004), </span><a class="ulink" href="http://ibis.inrialpes.fr/article533.html" target="_top"><span lang="" class="Hyperlink">Qualitative simulation of the initiation of
        sporulation in <span class="emphasis"><em>Bacillus
        subtilis</em></span></span></a><span lang="">,
        <span class="emphasis"><em>Bulletin of Mathematical Biology</em></span>, 66(2):261-300.
        </span></p></li><li class="listitem" style="list-style-type: disc"><p>D. Ropers, H. de Jong, M. Page, D. Schneider, J. Geiselmann
        (2006), <a class="ulink" href="http://ibis.inrialpes.fr/article742.html" target="_top">Qualitative simulation
        of the carbon starvation response in Escherichia coli</a>,
        <span class="emphasis"><em>Biosystems</em></span>, 84(2):124-152.</p></li><li class="listitem" style="list-style-type: disc"><p>J-A. Sepulchre, S. Reverchon, W. Nasser (2007), Modeling the
        onset of virulence in a pectinolytic bacterium, <span class="emphasis"><em>Journal of
        Theoretical Biology</em></span>, 44(2):239-257.</p></li><li class="listitem" style="list-style-type: disc"><p>A. Usseglio Viretta, M. Fussenegger (2004), Modeling the quorum
        sensing regulatory network of human-pathogenic <span class="emphasis"><em>Pseudomonas
        aeruginosa</em></span>, <span class="emphasis"><em>Biotechnology Progress</em></span>,
        20(3):670-678.</p></li></ul></div><p>An introduction to the use of GNA, with modeling tricks, can be
    found in:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p>G. Batt, B. Besson, P. Ciron, H. de Jong, E. Dumas, J.
          Geiselmann, R. Monte, P.T. Monteiro, M. Page, F. Rechenmann, D.
          Ropers, Genetic Network Analyzer: A Tool for the Qualitative
          Modeling and Simulation of Bacterial Regulatory Networks. In D.
          Thieffry et al (eds), <span class="emphasis"><em>Bacterial Molecular Networks:
          Methods and Protocols</em></span>, Human press, to appear.</p></li></ul></div><p>GNA uses the following open-source software: <a class="ulink" href="http://www.jgraph.com" target="_top">JGraph</a>, <a class="ulink" href="http://www.sat4j.org" target="_top">SAT4J</a>, and <a class="ulink" href="http://www2.cs.tum.edu/projects/cup/" target="_top">CUP</a>.The model-checker
    web server at INRIA runs <a class="ulink" href="http://nusmv.fbk.eu/" target="_top">NuSMV</a>.</p></div><div class="section" title="1.2.&nbsp;Projects using GNA"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.overview.projects"></a>1.2.&nbsp;Projects using GNA</h2></div></div></div><p>GNA has been used in several research projects, including:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><a class="ulink" href="http://www.cobios.net/" target="_top">COBIOS: Engineering and
        Control of Biological Systems: A New Way to Tackle Complex Diseases
        and Biotechnological Innovation</a> (2007-2009), <a class="ulink" href="http://cordis.europa.eu/nest/path_2005-2006.htm#bio" target="_top">NEST-2005-2
        PATHFINDER</a>.</p></li><li class="listitem" style="list-style-type: disc"><p>EC-MOAN: Scalable Modeling and Analysis Techniques to Study
        Emergent Cell Behavior; Understanding the E. coli Stress Response
        (2007-2009), <a class="ulink" href="http://cordis.europa.eu/nest/path_2005-2006.htm#complex" target="_top">NEST-2005-2
        PATHFINDER</a>.</p></li><li class="listitem" style="list-style-type: disc"><p>MetaGenoReg: Towards an Understanding of the Interrelations
        between Metabolic and Gene Regulation: E. coli Carbon Metabolism as a
        Test Case (2006-2009), <a class="ulink" href="http://www.agence-nationale-recherche.fr/appel-a-projet/21?lngAAPId=69" target="_top">ANR
        Biologie syst&eacute;mique (BIOSYS)</a>.</p></li><li class="listitem" style="list-style-type: disc"><p><a class="ulink" href="http://www.hygeiaweb.gr" target="_top">Hygeia: Hybrid Systems for
        Biochemical Network Modeling and Analysis</a> (2005-2007), <a class="ulink" href="http://www.cordis.lu/nest/adventure.htm" target="_top">NEST-2003-1
        ADVENTURE</a>.</p></li></ul></div></div></div><div class="chapter" title="Part 2. Using GNA"><div class="titlepage"><div><div><h2 class="title"><a name="iogma.gna.help.using"></a>Part 2. Using GNA</h2></div></div></div><div class="section" title="2.1.&nbsp;Defining a network"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.using.defining_network"></a>2.1.&nbsp;Defining a network</h2></div></div></div><p>The initial step in modeling a genetic regulatory network consists
    of the unambiguous definition of the network structure and the
    representation of this information in visual form. The genes, proteins,
    and interactions making up the network are obtained from the experimental
    literature and completed with plausible hypotheses. Notice that the
    collection of this information may take a sizeable amount of the total
    time invested in the modeling project, as it demands reflection on the
    precise biological questions to be addressed, followed by a careful
    evaluation of the available experimental data.</p><p>The network editor of GNA supports the modeler in defining and
    representing the network structure. It takes its inspiration from the
    visual modeling language <a class="ulink" href="http://link.aip.org/link/?CHAOEH/11/84/1" target="_top">MIM (Molecular Interaction
    Maps)</a> developed by Kohn and colleagues, and adheres as much as
    possible to the visualization standard proposed in the <a class="ulink" href="http://sbgn.org" target="_top">Systems Biology Graphical Notation (SBGN)</a>
    project. The network editor is built upon the IogmaNetwork library
    developed by <a class="ulink" href="http://www.genostar.com" target="_top">Genostar</a>.</p><div class="section" title="2.1.1.&nbsp;Define simple genetic regulatory network"><div class="titlepage"><div><div><h3 class="title"><a name="N1018A"></a>2.1.1.&nbsp;Define simple genetic regulatory network</h3></div></div></div><p>In order to illustrate the construction of a genetic regulatory
      network by means of the network editor, we consider a small network of
      three genes. Each gene codes for a protein that controls the expression
      of one or more target genes by binding to the promoter region. In
      particular, gene <span class="emphasis"><em>b</em></span> codes for a protein that
      activates the transcription of gene <span class="emphasis"><em>a</em></span>, while gene
      <span class="emphasis"><em>a</em></span> and gene <span class="emphasis"><em>c</em></span> code for proteins
      that form a heterodimer inhibiting the transcription of gene
      <span class="emphasis"><em>b</em></span>. In addition, protein A activates gene
      <span class="emphasis"><em>c</em></span>, while protein C inhibits the expression of its
      own gene.</p><p>The network can be created from the elements shown in the toolbar,
      among which we find genes and promoters, proteins, complex formation
      reactions, and activating and inhibiting interactions. Upon selecting a
      network element, the user is asked to specify the name of the gene or
      the protein. The position of the network elements in the editor can be
      changed by selecting an element and dragging it to another location
      using the mouse or the keyboard arrows. The network lay-out can be
      modified by adding or removing breakpoints in the lines representing the
      complex formation reactions and regulatory interactions. These
      breakpoints are added by means of mouse clicks while holding the
      <span class="keycap"><strong>Shift</strong></span> key down and can be moved like other network
      entities. The result for the example network is shown in <a class="xref" href="#figure.network" title="Fig.&nbsp;2.1.&nbsp;&nbsp;Schema of simple genetic regulatory network">Figure&nbsp;2.1</a>. To create reactions with multiple inputs
      and/or outputs, create first a simple reaction, then use the reaction
      tool again between the entity you want to add and the targeted part of
      the reaction (either its start, or its end).</p><div class="figure"><a name="figure.network"></a><p xmlns:fo="http://www.w3.org/1999/XSL/Format" xmlns:date="http://exslt.org/dates-and-times" class="title">Fig.&nbsp;2.1.&nbsp;&nbsp;Schema of simple genetic regulatory network</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen1.png" align="middle" alt="Schema of simple genetic regulatory network"></div></div></div><br class="figure-break"><p>The network thus constructed is saved in a so-called <span class="bold"><strong>project</strong></span>. You can save the project by choosing
      <span class="guimenuitem">Save project</span> in the <span class="guimenu">File</span>
      menu.When saving the schema, the lay-out is stored as well. You can also
      start a new project or load an existing project by choosing the
      corresponding options in the <span class="guimenu">File</span> menu.</p></div><div class="section" title="2.1.2.&nbsp;Define more complex regulatory networks"><div class="titlepage"><div><div><h3 class="title"><a name="N101CC"></a>2.1.2.&nbsp;Define more complex regulatory networks</h3></div></div></div><p>In <a class="xref" href="#figure.network" title="Fig.&nbsp;2.1.&nbsp;&nbsp;Schema of simple genetic regulatory network">Figure&nbsp;2.1</a> only a limited range of
      biochemical mechanisms involved in gene regulation was used, notably
      activation and inhibition of gene expression by transcription factors
      and the formation of protein complexes through dimerization. The network
      editor allows a much wider range of biochemical mechanisms to be
      described, as illustrated in <a class="xref" href="#figure.complex_network" title="Fig.&nbsp;2.2.&nbsp;&nbsp;Schema of a more complex genetic regulatory network">Figure&nbsp;2.2</a>.
      Examples include:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p>Enzymatic reactions (the reaction transforming E into F,
            catalyzed by D);</p></li><li class="listitem"><p>Post-translational modifications (the modification of B into
            B*, induced by an external signal);</p></li><li class="listitem"><p>Active degradation of proteins and mRNA (the degradation of
            C, regulated by B*).</p></li></ul></div><p>The types of network elements and biochemical reaction mechanisms
      supported by the network editor are shown in the menus and the toolbar
      (see <a class="xref" href="#iogma.gna.help.using.gna_functions" title="2.8.&nbsp;Functions of the graphical user interface">Section&nbsp;2.8</a>).</p><div class="figure"><a name="figure.complex_network"></a><p class="title">Fig.&nbsp;2.2.&nbsp;&nbsp;Schema of a more complex genetic regulatory network</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/network_complex.png" align="middle" alt="Schema of a more complex genetic regulatory network"></div></div></div><br class="figure-break"></div><div class="section" title="2.1.3.&nbsp;Reduction of regulatory network"><div class="titlepage"><div><div><h3 class="title"><a name="N101F6"></a>2.1.3.&nbsp;Reduction of regulatory network</h3></div></div></div><p><a class="xref" href="#figure.complex_network" title="Fig.&nbsp;2.2.&nbsp;&nbsp;Schema of a more complex genetic regulatory network">Figure&nbsp;2.2</a> gives an impression of
      the diversity of the elements composing a genetic regulatory network.
      For our purposes, it is convenient to have a more restricted view of a
      genetic regulatory network, as consisting of genes, gene products
      (proteins and mRNAs), and the direct and indirect regulation of gene
      expression exerted by the gene products. This results in a reduced
      network structure where boxes represent indirect regulatory interactions
      mediated by, for example, protein-protein interactions, metabolism, and
      signal transduction. The construction of these boxes is based on the
      assumption that the latter processes are typically fast on the
      time-scale of gene expression. A detailed explanation of this reduction
      step is beyond the scope of this tutorial, but more details can be found
      in publications listed in the <a class="link" href="#iogma.gna.help.overview.whatis_gna" title="1.1.&nbsp;What is GNA?">literature
      section</a>.</p><p>The reduction process is automatic and the user can see the result
      at any time by pressing the <span class="guibutton">Show reduced network</span>
      button in the toolbar. <a class="xref" href="#figure.reduced_network" title="Fig.&nbsp;2.3.&nbsp;&nbsp;Schema of reduced genetic regulatory network, obtained from Figure&nbsp;2.1">Figure&nbsp;2.3</a> shows
      the result for the simple three-gene network in <a class="xref" href="#figure.network" title="Fig.&nbsp;2.1.&nbsp;&nbsp;Schema of simple genetic regulatory network">Figure&nbsp;2.1</a>. The grey circles represent the proteins and
      the arrows the direct and indirect regulatory interactions controlling
      their synthesis and degradation. Notice that the reduction step has
      introduced a box mediating the influence of A and C on the expression of
      the gene encoding protein B. This influence is exerted through the
      heterodimerization of A and C, an interaction no longer explicitly
      represented but hidden in the box. The user can easily switch back and
      forth between the full and reduced network representation, thus making
      it possible to verify the pattern of indirect interactions induced by
      particular hypotheses on the biochemical mechanisms involved in the
      regulation of a gene.</p><div class="figure"><a name="figure.reduced_network"></a><p class="title">Fig.&nbsp;2.3.&nbsp;&nbsp;Schema of reduced genetic regulatory network, obtained from
        <a class="xref" href="#figure.network" title="Fig.&nbsp;2.1.&nbsp;&nbsp;Schema of simple genetic regulatory network">Figure&nbsp;2.1</a></p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/network_reduced.png" align="middle" alt="Schema of reduced genetic regulatory network, obtained from"></div></div></div><br class="figure-break"></div></div><div class="section" title="2.2.&nbsp;Building a model"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.using.building_model"></a>2.2.&nbsp;Building a model</h2></div></div></div><p>The <a class="link" href="#iogma.gna.help.using.defining_network" title="2.1.&nbsp;Defining a network">definition
    and reduction</a> of the genetic regulatory network under study gives
    rise to a schema summarizing the relevant genes, proteins, and
    interactions. In the next step, this schema is transformed into a GNA
    model of the network, by adding more information on the regulation of the
    genes. A GNA model has the form of a system of <span class="bold"><strong>piecewise-linear differential equations (PLDEs)</strong></span>,
    completed by a set of inequality constraints. The formulation of the model
    will be illustrated here, by means of the simple example introduced in the
    previous section.</p><p>The model-building process consists of four steps:</p><div class="orderedlist"><ol class="orderedlist" type="1"><li class="listitem"><p>Definition of state and input variables</p></li><li class="listitem"><p>Definition of parameters</p></li><li class="listitem"><p>Definition of state equations</p></li><li class="listitem"><p>Definition of inequality constraints</p></li></ol></div><p>The model-building process can be automated to some extent by means
    of a model-building wizard. Each of the above steps, and their integration
    into the model-building wizard, is discussed in detail below.</p><p>Most of the time, you will not build a model from scratch, but start
    from an earlier model. In that case, you choose <span class="guimenuitem">Open
    project...</span> in the <span class="guimenu">File</span> menu to load the
    earlier model you have decided to use as a template. You may import models
    from earlier GNA versions (text files with the extension <code class="filename">.gna</code>) by choosing <span class="guimenuitem">Import
    model...</span> in the <span class="guimenu">File</span> menu. It is also
    possible to import a model in SBML format, by choosing <span class="guimenuitem">Import
    model from SBML...</span> in the same <span class="guimenu">File</span>
    menu.</p><div class="section" title="2.2.1.&nbsp;Definition of state and input variables"><div class="titlepage"><div><div><h3 class="title"><a name="iogman.gna.using.building_model.definition_of_variables"></a>2.2.1.&nbsp;Definition of state and input variables</h3></div></div></div><p>In order to complete the network into a PLDE model, you have to
      click the <span class="guibutton">Model</span> tab in the left-upper corner of
      the application. GNA opens a desktop with two components: the <span class="bold"><strong>project tree</strong></span> and the <span class="bold"><strong>
      <span class="guilabel">Reduced network</span> </strong></span> window. The project
      tree lists the variables of the model as well as the initial conditions,
      atomic propositions and properties, while the <span class="guilabel">Reduced
      network</span> window shows the <a class="link" href="#figure.reduced_network" title="Fig.&nbsp;2.3.&nbsp;&nbsp;Schema of reduced genetic regulatory network, obtained from Figure&nbsp;2.1">network of direct and indirect genetic
      regulatory interactions</a> as obtained through reduction (see <a class="xref" href="#figure.reduced_network_window" title="Fig.&nbsp;2.4.&nbsp;&nbsp;Reduced network window with some examples of warnings">Figure&nbsp;2.4</a>).</p><p>The <span class="guilabel">Reduced network</span> window is also a way to
      check the consistency of the model definition with the network schema.
      It will graphically indicate the differences between both
      representations. The warnings do not impact the modeling process and can
      be ignored, but it is usually advisable to check if the differences are
      intentional or not. Warnings are produced in the following
      cases:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p>A network element does not exist as a model variable;</p></li><li class="listitem"><p>A variable created in the model has no corresponding element
            in the network;</p></li><li class="listitem"><p>The regulation of a variable differs between the network and
            the model (<span class="emphasis"><em>e.g.</em></span>, in <a class="xref" href="#figure.reduced_network_window" title="Fig.&nbsp;2.4.&nbsp;&nbsp;Reduced network window with some examples of warnings">Figure&nbsp;2.4</a> the regulation of C by
            A cannot be found in the state equation for C);</p></li><li class="listitem"><p>The equation for a state variable is invalid.</p></li></ul></div><p>The <span class="guilabel">Legend</span> tab allows you to
      quickly understand the meaning of the different entities of the reduced
      network. The <span class="guilabel">Reduced network</span> window will not be
      present if you work on a model for which no network has been defined
      previously.</p><div class="figure"><a name="figure.reduced_network_window"></a><p class="title">Fig.&nbsp;2.4.&nbsp;&nbsp;<span class="guilabel">Reduced network</span> window with some examples
        of warnings</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/reduced_network_window.png" align="middle" alt="Reduced network window with some examples of warnings"></div></div></div><br class="figure-break"><p>The genes in the network correspond to variables in the model,
      representing the concentrations of the proteins encoded by the genes.
      The variables are created by choosing the option <span class="guimenuitem">New
      variable</span> in the <span class="guimenu">Edit</span> menu. You can also
      press the <span class="mousebutton">right</span> button of your mouse when the
      cursor is on the model item in the <span class="guilabel">Project</span> window.
      In the remainder of this page, we will omit the shorthand commands for
      menu options, which are listed in the overview of <a class="link" href="#iogma.gna.help.using.gna_functions" title="2.8.&nbsp;Functions of the graphical user interface">GNA functions</a>. The
      creation of a variable is accomplished by specifying its name and leads
      to the opening of the <span class="guilabel">Variable</span> window. In the case
      of the example network, three variables are defined:
      <code class="varname">x_a</code>, <code class="varname">x_b</code>, and
      <code class="varname">x_c</code>. After their creation, variables can be renamed
      or deleted, by means of the options <span class="guimenuitem">Rename
      variable</span> and <span class="guimenuitem">Delete variable</span> in
      the <span class="guimenu">Edit</span> menu.</p><p>Two different types of variables are distinguished: <span class="bold"><strong>state variables</strong></span> and <span class="bold"><strong>input
      variables</strong></span>. While the former vary as a function of the other
      concentration variables in the model, the latter are determined by
      factors outside the scope of the system. In addition, GNA assumes that
      they are constant. In the <span class="guilabel">Variable</span> window, a
      variable can be defined as an input variable. This hides a number of
      fields in the <span class="guilabel">Variable</span> window that are irrelevant
      for input variables.</p></div><div class="section" title="2.2.2.&nbsp;Definition of parameters"><div class="titlepage"><div><div><h3 class="title"><a name="iogman.gna.using.building_model.definition_of_parameters"></a>2.2.2.&nbsp;Definition of parameters</h3></div></div></div><p>The <span class="guilabel">Variable</span> window also allows you to
      specify a number of parameters. Upon creation of a state or input
      variable, a <span class="bold"><strong>zero parameter</strong></span> and a
      <span class="bold"><strong>maximum parameter</strong></span> are automatically
      defined in the <span class="bold"><strong>parameter tree</strong></span> on the
      left. The zero and maximum parameters indicate a lower and upper bound
      for the concentration of the protein, respectively. By selecting these
      parameters in the parameter tree, the default name assigned by GNA can
      be changed. For the variable <code class="varname">x_c</code>, we have a zero
      parameter <code class="varname">zero_c</code> and a maximum parameter
      <code class="varname">max_c</code>, as shown in <a class="xref" href="#figure.variable_window_param" title="Fig.&nbsp;2.5.&nbsp;&nbsp;Variable window after creation of the parameters for the state variable x_c">Figure&nbsp;2.5</a> below.</p><div class="figure"><a name="figure.variable_window_param"></a><p class="title">Fig.&nbsp;2.5.&nbsp;&nbsp;<span class="guilabel">Variable</span> window after creation of the
        parameters for the state variable <code class="varname">x_c</code></p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen3.png" align="middle" alt="Variable window after creation of the parameters for the state variable x_c"></div></div></div><br class="figure-break"><p>The parameter tree may also contain a number of <span class="bold"><strong>threshold parameters</strong></span> for the variable. A
      threshold indicates the concentration above or below which a regulatory
      protein has an influence on the expression of the target gene. As a
      protein may be involved in several regulatory interactions, it may have
      several different thresholds The thresholds are created by pressing the
      <span class="guibutton">New</span> button of the tree menu, which asks the name
      of the threshold and inserts the newly created threshold into the tree.
      Once created, a threshold parameter can be deleted or renamed. In the
      case of <code class="varname">x_c</code>, we define thresholds
      <code class="varname">t_c1</code> and <code class="varname">t_c2</code>, arising from the
      two interactions in which the protein encoded by gene
      <span class="emphasis"><em>c</em></span> is involved. If two thresholds are equal, only a
      single threshold is defined (in other words, all thresholds are
      different).</p><p>Finally, if the variable is a state variable, you can specify
      so-called <span class="bold"><strong>synthesis</strong></span> and <span class="bold"><strong>degradation parameters</strong></span>. These indicate the rate
      of synthesis and degradation, respectively, of the protein whose
      concentration is represented by the variable. The synthesis and
      degradation parameters are specified in the parameter tree. The same
      buttons <span class="guibutton">New</span> and <span class="guibutton">Edit</span> allow
      the parameters to be created and maintained. Because there must always
      be at least one degradation parameter, GNA creates one by default. In
      the example, we introduce two synthesis parameters for
      <code class="varname">x_c</code>, <code class="varname">k_c1</code> and
      <code class="varname">k_c2</code>, representing two possible synthesis rates of
      protein C, respectively. In addition, we introduce the degradation
      parameter <code class="varname">g_c</code>. The result is shown in <a class="xref" href="#figure.variable_window_param" title="Fig.&nbsp;2.5.&nbsp;&nbsp;Variable window after creation of the parameters for the state variable x_c">Figure&nbsp;2.5</a>.</p></div><div class="section" title="2.2.3.&nbsp;Definition of state equations"><div class="titlepage"><div><div><h3 class="title"><a name="iogman.gna.using.building_model.definition_of_equations"></a>2.2.3.&nbsp;Definition of state equations</h3></div></div></div><p>Next, still in the <span class="guilabel">Variable</span> window, we add
      information on the regulatory logic of the synthesis and the degradation
      of the proteins. That is, we specify <span class="bold"><strong>state
      equations</strong></span>, which define the rate of change of the protein
      concentrations as a balance between the synthesis and the degradation
      rates of the protein. The state equations are entered in the
      <span class="guilabel">State equation</span> field, according to the <a class="link" href="#iogma.gna.help.appendix.model_syntax" title="3.1.&nbsp;Syntax of GNA model">syntax of GNA
      models</a> specified in the appendix.</p><p>The <span class="bold"><strong>synthesis term</strong></span> of the state
      equation is a sum of <span class="bold"><strong>step-function
      expressions</strong></span>, each weighted by a synthesis parameter. Recall
      that, in the case of protein C, we defined two synthesis rates,
      <code class="varname">k_c1</code> and <code class="varname">k_c2</code>, respectively.
      Suppose that the protein is not produced at all, if the concentration of
      protein C is above its threshold <code class="varname">t_c2</code>. That is, for
      <code class="varname">x_c &gt; t_c2</code>, negative autoregulation of the gene
      sets in. On the other hand, for <code class="varname">xc &lt; t_c2</code>, the
      protein is produced at a rate which is assumed to depend on the
      concentration of protein A. If the latter concentration is below the
      threshold <code class="varname">t_a2</code>, <span class="emphasis"><em>i.e.</em></span>
      <code class="varname">x_a &lt; t_a2</code>, then gene <span class="emphasis"><em>c</em></span> is
      expressed at a low rate <code class="varname">k_c1</code>. However, if
      <code class="varname">x_a &gt; t_a2</code>, then RNA polymerase is stabilized by
      protein A, and gene <span class="emphasis"><em>c</em></span> is expressed at a high rate
      <code class="varname">k_c1 + k_c2</code>. The following synthesis term captures
      this regulatory logic: <code class="varname">k_c1 * s-(x_c,t_c2) + k_c2 *
      s-(x_c,t_c2) * s+(x_a,t_a2)</code>, where
      <code class="varname">s-(x_c,t_c2)</code> and <code class="varname"> s+(x_a,t_a2)</code> are
      step-function expressions. That is, <code class="varname">s-(x_c,t_c2)</code>
      evaluates to 1, if <code class="varname">x_c &lt; t_c2</code>, whereas it
      evaluates to 0, if <code class="varname">x_c &gt; t_c2</code>. The case of
      <code class="varname">s+(x_a,t_a2)</code> goes analogously, while bearing in mind
      that <code class="varname">s+(x_a,t_a2) = 1 &ndash; s-(x_a,t_a2)</code>.</p><p>The <span class="bold"><strong>degradation term</strong></span> is defined
      analogously to the synthesis term, except that we demand that an
      unregulated, spontaneous degradation term is included, in order to
      ensure that the degradation rate of the protein is always positive. If
      we only have spontaneous degradation or growth dilution, as in the
      example of <a class="xref" href="#figure.variable_window_equa" title="Fig.&nbsp;2.6.&nbsp;&nbsp;Variable window after definition of the state equation for the variable x_c">Figure&nbsp;2.6</a>, then the
      degradation term is simply a degradation parameter multiplied by the
      state variable (<span class="emphasis"><em>e.g.</em></span>, <code class="varname">g_c *
      x_c</code>). In case the degradation of the protein is also regulated
      by other proteins, one or more step-function expressions, each weighted
      by a degradation parameter, are added to the degradation parameter
      accounting for spontaneous degradation. The resulting sum is multiplied
      by the state variable. For example, above a threshold concentration
      <code class="varname">t_b2</code>, protein B might favor degradation of protein C,
      by binding to it and thus presenting it to the proteolytic machinery.
      This can be represented by the degradation term <code class="varname">(g_c1 + g_c2 *
      s+(x_b,t_b2)) * x_c</code>, where <code class="varname">g_c1</code> is a low,
      spontaneous degradation rate, and <code class="varname">g_c1 + g_c2</code> a high,
      regulated degradation rate.</p><p>In general, the step-function expressions in the synthesis and the
      degradation terms may take many forms, depending on the number of
      proteins involved and the regulatory logic they implement. Further
      examples of step-function expressions, sometimes quite complex, can be
      found in the <a class="ulink" href="http://ibis.inrialpes.fr/article533.html" target="_top">model
      of the initiation of sporulation in <span class="emphasis"><em>B.
      subtilis</em></span></a> and the <a class="ulink" href="http://ibis.inrialpes.fr/article742.html" target="_top">model of the carbon
      starvation response in <span class="emphasis"><em>E. coli</em></span></a>.</p><p>The state equation is a differential equation whose righthand-side
      is the difference of the synthesis and degradation terms. In the
      example, the state equation for protein C is <code class="varname">d/dt x_c = k_c1 *
      s-(x_c,t_c2) + k_c2 * s-(x_c,t_c2) * s+(x_a,t_a2) &ndash; g_c * x_c</code>,
      as shown in <a class="xref" href="#figure.variable_window_equa" title="Fig.&nbsp;2.6.&nbsp;&nbsp;Variable window after definition of the state equation for the variable x_c">Figure&nbsp;2.6</a>. Once entered
      in the <span class="guilabel">State equation</span> field, the syntax and
      semantics of the state equation can be checked by clicking the
      <span class="guibutton">Check equation</span> button. It is then verified that
      the equation satisfies the <a class="link" href="#iogma.gna.help.appendix.model_syntax" title="3.1.&nbsp;Syntax of GNA model">syntax of GNA
      models</a> and that it is consistent with the parameter definitions
      in the other <span class="guilabel">Variable</span> windows.</p><div class="figure"><a name="figure.variable_window_equa"></a><p class="title">Fig.&nbsp;2.6.&nbsp;&nbsp;<span class="guilabel">Variable</span> window after definition of the
        state equation for the variable <code class="varname">x_c</code></p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen4.png" align="middle" alt="Variable window after definition of the state equation for the variable x_c"></div></div></div><br class="figure-break"></div><div class="section" title="2.2.4.&nbsp;Definition of inequality constraints"><div class="titlepage"><div><div><h3 class="title"><a name="iogma.gna.help.using.parameters_inequalities"></a>2.2.4.&nbsp;Definition of inequality constraints</h3></div></div></div><p>The simulation method does not require precise numerical values
      for the threshold, synthesis, and degradation parameters. Instead,
      inequality constraints on the parameter values are specified to predict
      the qualitative dynamics of the regulatory system (see the section on
      <a class="link" href="#iogma.gna.help.using.running_simulations" title="2.4.&nbsp;Running simulations">running
      simulations</a>). More precisely, GNA demands the user to order the
      so-called <span class="bold"><strong>threshold parameters</strong></span> and
      <span class="bold"><strong>focal parameters</strong></span>. We briefly explain
      the biological meaning of these parameters and their ordering by means
      of the example.</p><p>Protein C has two thresholds, <code class="varname">t_c1</code> and
      <code class="varname">t_c2</code>. Assume that the protein influences the
      expression of gene <span class="emphasis"><em>b</em></span> at lower concentrations than
      it influences the expression of its own gene. This gives rise to the
      inequality <code class="varname">t_c1 &lt; t_c2</code>. By default,
      <code class="varname">zero_c &lt; t_c1</code> and <code class="varname">t_c2 &lt;
      max_c</code>. The thresholds divide the phase space into regions in
      which the step functions evaluate to either 0 or 1, so that the state
      equations there reduce to linear and uncoupled differential equations.
      Inside a region, each protein concentration monotonically converges
      towards a so-called <span class="bold"><strong>focal parameter</strong></span>,
      given by a quotient of synthesis and degradation parameters of the
      protein. The state equation above, for instance, implies that in a
      region in which <code class="varname">x_a</code> lies below
      <code class="varname">t_a2</code>, and <code class="varname">x_c</code> below
      <code class="varname">t_c2</code>, <code class="varname">x_c</code> converges towards the
      focal parameter <code class="varname">k_c2/g_c</code>. The behavior of the system
      on the threshold planes separating the regions can be determined from
      the behavior in the adjacent regions, as explained in the <a class="link" href="#iogma.gna.help.overview.whatis_gna" title="1.1.&nbsp;What is GNA?">publications on the
      qualitative simulation method</a>.</p><p>The possible focal parameters in the different regions of the
      phase space can be ordered with respect to the threshold parameters.
      Intuitively speaking, this defines the strength of gene expression in
      the regions of the phase space in a qualitative way, on the scale of
      ordered threshold concentrations. In the case of variable
      <code class="varname">x_c</code>, the possible focal parameters are 0,
      <code class="varname">k_c1/g_c</code>, and <code class="varname">(k_c1 + k_c2)/g_c</code>.
      At the highest expression level, corresponding to the focal parameter
      <code class="varname">(k_c1 + k_c2)/g_c</code>, protein C should be able to
      repress its own synthesis. This implies that <code class="varname">t_c2 &lt; (k_c1 +
      k_c2)/g_c &lt; max_c</code>. In addition, we set <code class="varname">t_c1 &lt;
      k_c1/g_c &lt; t_c2</code>.</p><p>The ordering of the threshold and focal parameters, and thus the
      specification of the inequality constraints, is achieved in the
      <span class="guilabel">Parameter ordering</span> field. It contains an axis on
      which the previously defined threshold, zero, and maximum parameters, as
      well as the focal parameters automatically generated from the state
      equation are positioned. The elements of the list can be ordered by
      means of the mouse, by dragging them up and down the axis. Elements can
      also be moved out of the ordered list, by dropping them left from the
      zero parameter. Notice that this results in a total order on the
      threshold and focal parameters, contrary to what was the case in
      previous versions of GNA (in which focal parameters did not need to be
      totally ordered). As a consequence, when models prepared with previous
      versions of GNA (version 6.0 or below) are loaded in GNA 8, the program
      issues a warning that the parameter order is not total and appends the
      unordered focal parameters at the left of the zero parameter. <a class="xref" href="#figure.variable_window_inequal" title="Fig.&nbsp;2.7.&nbsp;&nbsp;Variable window after definition of the inequality constraints for the variable x_c">Figure&nbsp;2.7</a> shows the ordered threshold
      and focal parameters for the variable <code class="varname">x_c</code>.</p><div class="figure"><a name="figure.variable_window_inequal"></a><p class="title">Fig.&nbsp;2.7.&nbsp;&nbsp;<span class="guilabel">Variable</span> window after definition of the
        inequality constraints for the variable <code class="varname">x_c</code></p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen5.png" align="middle" alt="Variable window after definition of the inequality constraints for the variable x_c"></div></div></div><br class="figure-break"><p>After you have specified the inequality constraints, the GNA model
      of the genetic regulatory network is complete. You can save it by means
      of <span class="guimenuitem">Save project</span> or <span class="guimenuitem">Save project
      as...</span> in the <span class="guimenu">File</span> menu. The project is
      saved in an XML file with the extension <code class="filename">.xml</code> or <code class="filename">.gnaml</code>. You may also decide to export the
      model to a file with the <code class="filename">.gna</code>
      extension (former GNA files) by the means of <span class="guimenuitem">Export
      model...</span> in the <span class="guimenu">File</span> menu, or export it
      to SBML using <span class="guimenuitem">Export to SBML...</span>.</p><p>Examples of complete model files can be consulted in the <code class="filename">samples</code> directory of the GNA
      distribution.</p></div><div class="section" title="2.2.5.&nbsp;Definition of the model through the wizard"><div class="titlepage"><div><div><h3 class="title"><a name="N104C9"></a>2.2.5.&nbsp;Definition of the model through the wizard</h3></div></div></div><p>The first three steps of building a model (<a class="link" href="#iogman.gna.using.building_model.definition_of_variables" title="2.2.1.&nbsp;Definition of state and input variables">definition
      of state and input variables</a>, <a class="link" href="#iogman.gna.using.building_model.definition_of_parameters" title="2.2.2.&nbsp;Definition of parameters">definition
      of parameters</a> and <a class="link" href="#iogman.gna.using.building_model.definition_of_equations" title="2.2.3.&nbsp;Definition of state equations">definition
      of state equations</a>) can also be handled through a <span class="bold"><strong>wizard</strong></span> that allows you to semi-automatically
      complete these steps. The model definition wizard is available if you
      have previously defined a network. It is launched the first time you
      switch from the network to the model editor mode. Even if the wizard is
      capable of dealing with only a resttricted range of regulating
      mechanisms, and thus may not give correct results for complex networks,
      it is usually worth the effort to run it in order to prepare a basic -
      perhaps partial - model that will facilitate and speed up the rest of
      the modeling process.</p><p>The first step of the wizard allows you to decide which variables
      will be created in the model (see <a class="xref" href="#figure.wizard1" title="Fig.&nbsp;2.8.&nbsp;&nbsp;Wizard window for the definition of state and input variables">Figure&nbsp;2.8</a>).
      The default proposal is based on the following rules:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p>Genes and external signals will be used to create
            corresponding model variables;</p></li><li class="listitem"><p>Among these variables, those regulated by other variables
            will be tagged as state variables;</p></li><li class="listitem"><p>Other variables, not regulated, will be tagged as input
            variables and will not have a state equation associated. This will
            allways be the case for external signals.</p></li></ul></div><p>You can change this default if it does not correspond
      to what you want, as well as the name of the model variables.</p><div class="figure"><a name="figure.wizard1"></a><p class="title">Fig.&nbsp;2.8.&nbsp;&nbsp;<span class="guilabel">Wizard</span> window for the definition of state
        and input variables</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/wizard1.png" align="middle" alt="Wizard window for the definition of state and input variables"></div></div></div><br class="figure-break"><p>The second step of the wizard consists in defining an equation for
      each state variable you want to take into account (you may decide not to
      let the wizard algorithm generate an equation for some of the variables
      and define them yourself later on). For a specific state variable, the
      wizard takes all regulating variables into account to generate a
      consistent equation. You can specify various parameters, such as the
      number of synthesis and degradation parameters, the variables implied
      for each of these parameters, and the regulatory logic applied (you can
      choose between AND, NAND, OR and NOR). This is illustrated in <a class="xref" href="#figure.wizard2" title="Fig.&nbsp;2.9.&nbsp;&nbsp;Wizard window for the definition of the state equation of B">Figure&nbsp;2.9</a>. When the options fit your needs, click on
      the <span class="guibutton">Next</span> button to reach the next step of the
      model-building-process.</p><div class="figure"><a name="figure.wizard2"></a><p class="title">Fig.&nbsp;2.9.&nbsp;&nbsp;<span class="guilabel">Wizard</span> window for the definition of the
        state equation of B</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/wizard2.png" align="middle" alt="Wizard window for the definition of the state equation of B"></div></div></div><br class="figure-break"><p>When working with the wizard, you can go back one or more steps,
      or even cancel the process. When you have successfully completed all
      steps of the wizard, you still have to <a class="link" href="#iogma.gna.help.using.parameters_inequalities" title="2.2.4.&nbsp;Definition of inequality constraints">order the
      parameters</a> for each variable. After this you can start <a class="link" href="#iogma.gna.help.using.building_initial_conditions" title="2.3.&nbsp;Specifying initial conditions">specifying
      initial conditions</a> and continue simulating and analyzing the
      model.</p></div></div><div class="section" title="2.3.&nbsp;Specifying initial conditions"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.using.building_initial_conditions"></a>2.3.&nbsp;Specifying initial conditions</h2></div></div></div><p>As explained in the section on <a class="link" href="#iogma.gna.help.using.building_model" title="2.2.&nbsp;Building a model">model building</a>, the
    threshold parameters divide the phase space into regions. These regions
    can be further refined by means of the focal parameters, so as to arrive
    at a partition of the phase space consisting of <span class="bold"><strong>domains</strong></span> in which the concentration variables have a
    unique derivative sign pattern (see the <a class="link" href="#iogma.gna.help.overview.whatis_gna" title="1.1.&nbsp;What is GNA?">publications on the
    simulation method</a> for mathematical details). Stated in a different
    way, the resulting partition guarantees that the system behaves in a
    qualitatively-identical way everywhere in a domain.</p><p>Before starting the simulation, you must specify <span class="bold"><strong>initial conditions</strong></span> in the form of an initial domain
    or an initial set of domains. This is achieved by selecting the
    <span class="guimenuitem">New initial conditions</span> in the
    <span class="guisubmenu">New</span> submenu of the <span class="guimenu">Edit</span> menu.
    This opens the <span class="guilabel">Initial conditions</span> window, in which
    you can specify (strict) lower and upper bounds on the variables by
    drawing appropriate intervals by means of the mouse. The lower and upper
    bounds consist of threshold or focal parameters. Opening the window also
    adds the new initial conditions to the project tree.</p><p>Suppose that we want to simulate the example network of <a class="xref" href="#figure.network" title="Fig.&nbsp;2.1.&nbsp;&nbsp;Schema of simple genetic regulatory network">Figure&nbsp;2.1</a> from initial conditions <code class="varname">zero_a &lt;=
    x_a &lt; t_a1</code>, <code class="varname">zero_b &lt;= x_b &lt; t_b</code>, and
    <code class="varname">zero_c &lt;= x_c &lt; t_c1</code>. That is, we assume that the
    concentrations of the variables are between zero and their first
    threshold. <a class="xref" href="#figure.initial_conditions" title="Fig.&nbsp;2.10.&nbsp;&nbsp;Initial conditions window after definition of the initial conditions">Figure&nbsp;2.10</a> shows the
    corresponding <span class="guilabel">Initial conditions</span> window.</p><div class="figure"><a name="figure.initial_conditions"></a><p class="title">Fig.&nbsp;2.10.&nbsp;&nbsp;<span class="guilabel">Initial conditions</span> window after definition
      of the initial conditions</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen6.png" align="middle" alt="Initial conditions window after definition of the initial conditions"></div></div></div><br class="figure-break"><p>The initial conditions used by GNA are saved in the same project
    file as the model. However, you can also import and export initial
    conditions separately as files with the extension <code class="filename">.dat</code>, compatible with earlier versions of
    GNA. To this end, use <span class="guimenuitem">Import initial
    conditions...</span> and <span class="guimenuitem">Export initial
    conditions...</span> in the <span class="guimenu">File</span> menu.</p><p>Examples of other initial conditions files can be consulted in the
    <code class="filename">samples</code> directory of the GNA
    distribution.</p></div><div class="section" title="2.4.&nbsp;Running simulations"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.using.running_simulations"></a>2.4.&nbsp;Running simulations</h2></div></div></div><p>In every domain of the phase space, the system behaves in a
    qualitatively identical way, in the sense that the concentration variables
    have a unique derivative sign pattern in the domain (see the section on
    the <a class="link" href="#iogma.gna.help.using.building_initial_conditions" title="2.3.&nbsp;Specifying initial conditions">specification
    of initial conditions</a>). For this reason, a domain can be seen as
    representing a <span class="bold"><strong>qualitative state</strong></span> of the
    system. A <span class="bold"><strong>transition</strong></span> between qualitative
    states occurs, if the concentration variables have evolved in such a way
    that the system leaves one domain and enters another. The simulation
    algorithm of GNA recursively computes all qualitative states that are
    reachable through one or more transitions from the qualitative state(s)
    corresponding to the initial domain(s). This results in a <span class="bold"><strong>state transition graph</strong></span>, summarizing the qualitative
    dynamics of the system. The graph contains <span class="bold"><strong>qualitative steady states</strong></span> or <span class="bold"><strong>qualitative cycles</strong></span>, which may correspond to steady
    states and limit cycles, respectively, of the underlying piecewise-linear
    differential equations. Notice that, due to the mathematical techniques
    used to describe the dynamics of the system in threshold planes, the
    steady states located on the thresholds of the concentration variables may
    be sets instead of points (see the <a class="link" href="#iogma.gna.help.overview.whatis_gna" title="1.1.&nbsp;What is GNA?">GNA publications</a> for
    mathematical details).</p><p>Once you have defined a model of the genetic regulatory network and
    appropriate initial conditions, GNA allows you to perform a qualitative
    simulation of the network. First, you need to define the simulation
    parameters, either by choosing <span class="guisubmenu">Specify simulation</span>
    in the <span class="guimenu">Analysis</span> menu, or by pressing the corresponding
    button in the <span class="guilabel">Initial conditions</span> window. This opens
    the <span class="guilabel">Properties</span> page of the
    <span class="guilabel">Simulation</span> window, in which you can set the time-out
    parameter and the maximum number of states of the state transition graph
    that will be displayed. In addition, you can specify whether the
    simulation should be constrained to the initial conditions only or not.
    That is, by checking this option GNA only computes the transitions between
    the qualitative states corresponding to domains included in the initial
    conditions, and does not consider transitions to other states. This gives
    rise to a reduced state transition graph, summarizing the qualitative
    dynamics of the system in a specific region of the phase space, for
    instance the dynamics of the system around a steady state.</p><p>The actual simulation is launched by choosing
    <span class="guibutton">Run</span> in the <span class="guilabel">Simulation</span> window
    or in the <span class="guimenu">Analysis</span> menu. The program displays run-time
    statistics in the <span class="guilabel">Simulation</span> window, as shown in
    figure <a class="xref" href="#figure.simulation_window" title="Fig.&nbsp;2.11.&nbsp;&nbsp;Simulation window after running the model of the genetic regulatory network in Figure&nbsp;2.1">Figure&nbsp;2.11</a>, and allows you to
    abort the simulation.</p><div class="figure"><a name="figure.simulation_window"></a><p class="title">Fig.&nbsp;2.11.&nbsp;&nbsp;<span class="guilabel">Simulation</span> window after running the model of
      the genetic regulatory network in <a class="xref" href="#figure.network" title="Fig.&nbsp;2.1.&nbsp;&nbsp;Schema of simple genetic regulatory network">Figure&nbsp;2.1</a></p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen7.png" align="middle" alt="Simulation window after running the model of the genetic regulatory network in"></div></div></div><br class="figure-break"><p>A qualitative simulation can also be launched from the console using
    the MS-DOS, Linux, Unix, or MacOS command:</p><pre class="screen">gna [-p] [-v] [-q] [-e &lt;plugin-directory&gt;]
    [-f] [-i &lt;file&gt;] &lt;project/model&gt;</pre><p>The argument <code class="option">project/model</code> is mandatory and refers
    to the project file (extension <code class="filename">.gnaml</code>) or the model file used in previous
    GNA releases (extension <code class="filename">.gna</code>). The
    <code class="option">-p</code> option (also accessible through
    <code class="option">--presentation-mode</code>) is set to <em class="parameter"><code>user</code></em> when GNA is started from the command line
    by the user, and to <em class="parameter"><code>system</code></em> when GNA
    is called from within another application. The trace level determines the
    amount of trace information generated during the simulation, and can be
    adjusted through the <code class="option">-v</code> (<code class="option">--verbose</code>) and
    <code class="option">-q</code> (<code class="option">--quiet</code>) options. If the project
    does not contain any initial conditions, and the <code class="option">-i</code>
    (<code class="option">--initial-conditions</code>) option is not specified, the
    regulatory network is simulated for all possible initial conditions. That
    is, GNA generates a state transition graph containing all possible
    qualitative states of the system. (However, note that in enumerating the
    possible qualitative states, GNA omits any qualitative states associated
    with domains in which an input variable assumes a threshold value).</p><p>As will be explained in section <a class="xref" href="#iogma.gna.help.using.building_property" title="2.7.&nbsp;Checking properties of the state transition graph">Section&nbsp;2.7</a>, the simulation
    results can be analyzed by means of a model checker. The command
    <span class="command"><strong>gna</strong></span> is able to generate model-checker specific output,
    as defined by customized plug-ins. The option <code class="option">-e</code>
    (<code class="option">--export</code>) provides the plug-in directory to be used. If
    the simulation results are in the form of an explicit state transition
    graph, one can additionally filter the instantaneous states of the state
    transition graph through the <code class="option">-f</code> option
    (<code class="option">--filter-instantaneous-states</code>).</p><p>The command</p><pre class="screen">gna --help</pre><p>displays the meaning of the arguments.</p></div><div class="section" title="2.5.&nbsp;Interpreting simulation results"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.using.interpreting_results"></a>2.5.&nbsp;Interpreting simulation results</h2></div></div></div><p>The <span class="guilabel">Results</span> page in the
    <span class="guilabel">Simulation</span> window summarizes the results obtained for
    a qualitative simulation (<a class="xref" href="#figure.simulation_window_results" title="Fig.&nbsp;2.12.&nbsp;&nbsp;Simulation window displaying results of a qualitative simulation of the genetic regulatory network in Figure&nbsp;2.1: Qualitative steady states, qualitative cycles, and their attraction sets">Figure&nbsp;2.12</a>). It displays the
    qualitative steady states and qualitative cycles that are reachable from
    the initial qualitative states, as well as the <span class="bold"><strong>attraction set</strong></span> of each qualitative steady state or
    qualitative cycle (<span class="emphasis"><em>i.e.</em></span>, the set of qualitative
    states from which the attractor is reachable). The
    <span class="guilabel">Results</span> page in the <span class="guilabel">Simulation</span>
    window also gives the name of the <span class="guilabel">Transition graph</span>
    window, containing the state transition graph resulting from the
    simulation (<a class="xref" href="#figure.simulation_window_graph" title="Fig.&nbsp;2.13.&nbsp;&nbsp;Results of a qualitative simulation of the genetic regulatory network in Figure&nbsp;2.1: Fragment of the state transition graph">Figure&nbsp;2.13</a>). In the
    example, the state transition graph consists of 50 qualitative states,
    associated with domains lying between threshold or focal planes
    (regulatory domains), or domains lying on one or more threshold or focal
    planes (switching domains). In order to graphically distinguish the
    qualitative states associated with regulatory and switching domains, the
    former have an emphasized border. In addition, qualitative steady states
    are shaded.</p><div class="figure"><a name="figure.simulation_window_results"></a><p class="title">Fig.&nbsp;2.12.&nbsp;&nbsp;<span class="guilabel">Simulation</span> window displaying results of a
      qualitative simulation of the genetic regulatory network in <a class="xref" href="#figure.network" title="Fig.&nbsp;2.1.&nbsp;&nbsp;Schema of simple genetic regulatory network">Figure&nbsp;2.1</a>: Qualitative steady states, qualitative
      cycles, and their attraction sets</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen8.png" align="middle" alt="Simulation window displaying results of a qualitative simulation of the genetic regulatory network in : Qualitative steady states, qualitative cycles, and their attraction sets"></div></div></div><br class="figure-break"><div class="figure"><a name="figure.simulation_window_graph"></a><p class="title">Fig.&nbsp;2.13.&nbsp;&nbsp;Results of a qualitative simulation of the genetic regulatory
      network in <a class="xref" href="#figure.network" title="Fig.&nbsp;2.1.&nbsp;&nbsp;Schema of simple genetic regulatory network">Figure&nbsp;2.1</a>: Fragment of the state
      transition graph</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen9.png" align="middle" alt="Results of a qualitative simulation of the genetic regulatory network in : Fragment of the state transition graph"></div></div></div><br class="figure-break"><p>The state transition graph contains more detailed information on the
    qualitative dynamics of the system, which can be analyzed by means of the
    graphical user interface. First, by hovering the mouse cursor on a
    qualitative state, the properties characterizing the state, notably its
    location in the phase space, are displayed in textual form. Press the
    <span class="keycap"><strong>F2</strong></span> key if you want to focus on it. Second, you can select
    a path in the state transition graph and use the option
    <span class="guimenuitem">Display variables in selected path</span> in the
    <span class="guimenu">Results</span> menu to show the qualitative evolution of the
    protein concentrations along the selected path. The states are selected by
    means of the mouse, while pressing the <span class="keycap"><strong>Ctrl</strong></span> key. A click on a selected state, again
    while pressing the <span class="keycap"><strong>Ctrl</strong></span> key, allows
    this state to be deselected. In <a class="xref" href="#figure.simulation_window_path" title="Fig.&nbsp;2.14.&nbsp;&nbsp;Qualitative evolution of the variables x_a, x_b, and x_c in a path selected in the state transition graph of Figure&nbsp;2.13">Figure&nbsp;2.14</a>, you can see a path selected in
    the state transition graph of <a class="xref" href="#figure.simulation_window_graph" title="Fig.&nbsp;2.13.&nbsp;&nbsp;Results of a qualitative simulation of the genetic regulatory network in Figure&nbsp;2.1: Fragment of the state transition graph">Figure&nbsp;2.13</a>. Notice that the order in
    which the states have been selected determines the order in the
    concentration profiles shown. If the selected order is consistent with the
    successor relations in the state transition graph, then the path can be
    interpreted as a temporal order of events. More precisely, the
    concentration profile then shows how the derivative signs of the variables
    evolve over time, giving rise to a sequence of maxima and minima for each
    of the variables. This information can be straightforwardly compared with
    available gene expression data. Similarly, the option <span class="guimenuitem">Display
    focal sets in selected path</span> in the
    <span class="guimenu">Results</span> menu shows the qualitative evolution of the
    values towards which the state variables locally converge.</p><div class="figure"><a name="figure.simulation_window_path"></a><p class="title">Fig.&nbsp;2.14.&nbsp;&nbsp;Qualitative evolution of the variables
      <code class="varname">x_a</code>,<code class="varname"> x_b</code>, and
      <code class="varname">x_c</code> in a path selected in the state transition graph
      of <a class="xref" href="#figure.simulation_window_graph" title="Fig.&nbsp;2.13.&nbsp;&nbsp;Results of a qualitative simulation of the genetic regulatory network in Figure&nbsp;2.1: Fragment of the state transition graph">Figure&nbsp;2.13</a></p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen10.png" align="middle" alt="Qualitative evolution of the variables x_a, x_b, and x_c in a path selected in the state transition graph of"></div></div></div><br class="figure-break"><p>For more complex genetic regulatory networks, the state transition
    graph produced by GNA may contain hundreds or thousands of states.
    Depending on the display limit set in the <span class="guilabel">Properties</span>
    page of the <span class="guilabel">Simulation</span> window, only part of the state
    transition graph is then shown. The graphical user interface provides a
    few functions to analyze this graph. First, the zoom functions in the
    <span class="guimenu">Results</span> menu - more conveniently accessed through the
    <span class="mousebutton">right</span> mouse button (see the overview of <a class="link" href="#iogma.gna.help.using.gna_functions" title="2.8.&nbsp;Functions of the graphical user interface">GNA functions</a>) - allow
    you to increase or decrease the visible part of the graph. Second, you can
    select a subset of the state transition graph, display it in a separate
    tab, and expand it using several levels of successors and/or predecessors.
    Third, a useful option is to reduce the graph by eliminating qualitative
    states that correspond to domains that are instantaneously traversed, and
    therefore have no practical relevance. This option, called
    <span class="guimenuitem">Filter instantaneous states</span> in the
    <span class="guimenu">Results</span> menu, eliminates 32 of the 50 qualitative
    states of the state transition graph in our example. Fourth, the
    <span class="guilabel">Search states</span> page in the
    <span class="guilabel">Simulation</span> window allows you to identify parts of the
    state transition graph that satisfy certain search criteria. Finally, once
    a single state is selected it is possible to navigate through the
    transition graph using the keyboard arrows (maintain the <span class="keycap"><strong>Alt</strong></span> key pressed to add new states to the
    selection).</p><p>For really large models, the above functions may not be sufficient,
    and other solutions need to be tried. GNA offers two functionalities that
    are of particular interest for the analysis of large state transition
    graphs. First, you can export your graph to a model-checking tool and
    further analyze its properties by formulating queries in a temporal logic
    (see <a class="link" href="#iogma.gna.help.using.building_property" title="2.7.&nbsp;Checking properties of the state transition graph">section on
    checking properties of the state transition graph</a>). Second, instead
    of generating the entire state transition graph, you can use the attractor
    search option (see the <a class="link" href="#iogma.gna.help.using.attractors" title="2.6.&nbsp;Searching for attractors">section on searching for
    attractors</a>).</p></div><div class="section" title="2.6.&nbsp;Searching for attractors"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.using.attractors"></a>2.6.&nbsp;Searching for attractors</h2></div></div></div><p>A <a class="link" href="#iogma.gna.help.using.running_simulations" title="2.4.&nbsp;Running simulations">qualitative
    simulation</a> results in a state transition graph, consisting of
    qualitative states and transitions between qualitative states. When
    dealing with models of complex genetic regulatory networks, the state
    transition graph may become too large to be handled by the <a class="link" href="#iogma.gna.help.using.interpreting_results" title="2.5.&nbsp;Interpreting simulation results">graph analysis
    functions</a> of GNA. In many cases it is not necessary to generate the
    state transition graph and it suffices to know which are the attractors of
    the system, that is, the qualitative steady states and qualitative cycles.
    The aim of the <span class="bold"><strong>attractor search</strong></span> function
    of GNA is to locate the qualitative steady states of the system, without
    generating the state transition graph.</p><p>The <span class="bold"><strong>stability</strong></span> of a qualitative
    steady state corresponds to the stability of a steady state, or more
    generally a steady state set, of the underlying piecewise-linear
    differential equations. The stability can be determined from the local
    topology of the state transition graph (see <a class="link" href="#iogma.gna.help.overview.whatis_gna" title="1.1.&nbsp;What is GNA?">GNA publications</a>).
    While a steady state not located on a threshold plane is always
    asymptotically stable, the stability of a steady state or steady state set
    located on a threshold plane can often be inferred from the incoming and
    outgoing transitions in the state transition graph. More precisely, if
    there is an incoming transition from all neighboring qualitative states,
    and no outgoing transition, then the equilibrium set is <span class="bold"><strong>asymptotically stable</strong></span>. On the other hand, if there
    is at least one outgoing transition, then the steady state is <span class="bold"><strong>unstable</strong></span>. In the other cases, the stability of the
    steady state cannot be determined from the topology of the state
    transition graph alone, and further mathematical analysis is required. The
    above notions of stability can be generalized to the steady state set
    sometimes occurring in piecewise-linear models (see <a class="link" href="#iogma.gna.help.overview.whatis_gna" title="1.1.&nbsp;What is GNA?">GNA
    publications</a>).</p><p>Once you have defined a model of the genetic regulatory network, GNA
    allows you to search for the steady states of the network. First you need
    to specify the search parameters, by choosing <span class="guimenuitem">Search
    attractors...</span> in the <span class="guimenu">Analysis</span> menu. This
    opens the <span class="guilabel">Properties</span> page of the <span class="guilabel">Attractor
    search</span> window, in which you can set the time-out parameter. The
    actual search for steady states is launched by choosing
    <span class="guibutton">Run</span> in the <span class="guilabel">Attractor search</span>
    window. The program displays run-time statistics in the
    <span class="guilabel">Attractor search</span> window and allows you to abort the
    attractor search.</p><div class="figure"><a name="figure.attractor_search"></a><p class="title">Fig.&nbsp;2.15.&nbsp;&nbsp;<span class="guilabel">Attractor search</span> window showing the results
      of the search for qualitative steady states in the model of the genetic
      regulatory network in <a class="xref" href="#figure.network" title="Fig.&nbsp;2.1.&nbsp;&nbsp;Schema of simple genetic regulatory network">Figure&nbsp;2.1</a>. The system has
      a single, asymptotically stable steady state. The attractor search also
      generates new initial conditions, consisting of the qualitative steady
      states that were found.</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen12.png" align="middle" alt="Attractor search window showing the results of the search for qualitative steady states in the model of the genetic regulatory network in . The system has a single, asymptotically stable steady state. The attractor search also generates new initial conditions, consisting of the qualitative steady states that were found."></div></div></div><br class="figure-break"><p>The <span class="guilabel">Results</span> page in the <span class="guilabel">Attractor
    search</span> window summarizes the results obtained (<a class="xref" href="#figure.attractor_search" title="Fig.&nbsp;2.15.&nbsp;&nbsp;Attractor search window showing the results of the search for qualitative steady states in the model of the genetic regulatory network in Figure&nbsp;2.1. The system has a single, asymptotically stable steady state. The attractor search also generates new initial conditions, consisting of the qualitative steady states that were found.">Figure&nbsp;2.15</a>). More particularly, the page
    contains a list of qualitative steady states, together with an indication
    of their stability (<span class="bold"><strong>asymptotically
    stable</strong></span>, <span class="bold"><strong>unstable</strong></span>, or <span class="bold"><strong>undetermined</strong></span>). Sometimes several adjacent
    qualitative steady states correspond to a single steady state set in the
    underlying piecewise-linear model. In this case, GNA groups the
    qualitative steady states together. If GNA is not able to determine all
    steady states or group them together (a computationally expensive step)
    before the time-out is reached, the partial results are presented.</p><p>Each qualitative steady state is also added as an initial condition
    in the project tree. By performing a <a class="link" href="#iogma.gna.help.using.running_simulations" title="2.4.&nbsp;Running simulations">qualitative simulation
    constrained to neighborhood</a> of the initial condition corresponding
    to a qualitative steady state, you can get an idea of the topology of the
    state transition graph around this state. This is especially useful if the
    stability of the steady state is left undetermined by GNA. <a class="xref" href="#figure.attractor_search" title="Fig.&nbsp;2.15.&nbsp;&nbsp;Attractor search window showing the results of the search for qualitative steady states in the model of the genetic regulatory network in Figure&nbsp;2.1. The system has a single, asymptotically stable steady state. The attractor search also generates new initial conditions, consisting of the qualitative steady states that were found.">Figure&nbsp;2.15</a> shows that the example model only has
    a single qualitative steady state, which is not located on a threshold
    plane and therefore asymptotically stable. The initial conditions
    <code class="varname">SteadyState1</code> are generated.</p><p>An attractor search can also be launched from the console using an
    MS-DOS, Linux, Unix, or MacOS command, like for a <a class="link" href="#iogma.gna.help.using.running_simulations" title="2.4.&nbsp;Running simulations">qualitative
    simulation</a>:</p><pre class="screen">gna-attractor &lt;project/model&gt;</pre><p>The argument <code class="option">project/model</code> refers to the file with
    the model. Typing</p><pre class="screen">gna-attractor --help</pre><p>in the console displays the meaning of the arguments.</p></div><div class="section" title="2.7.&nbsp;Checking properties of the state transition graph"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.using.building_property"></a>2.7.&nbsp;Checking properties of the state transition graph</h2></div></div></div><p>Another way to deal with large state transition graphs is to analyze
    the graph by means of formal verification techniques, implemented in
    computer tools called model checkers. Examples of the kind of properties
    one might want to test are "<span class="emphasis"><em>It is possible for a state with an
    <code class="varname">x_a</code> concentration equal to <code class="varname">t_xa2</code> to
    occur</em></span>" or "<span class="emphasis"><em>If the system is in the state
    corresponding to the initial conditions <code class="varname">zero_a &lt;= x_a &lt;
    t_a1</code>, <code class="varname">zero_b &lt;= x_b &lt; t_b</code>, and
    <code class="varname">zero_c &lt;= x_c &lt; t_c1</code>, then it necessarily reaches
    a steady state</em></span>".</p><p>The use of the model-checking tools involves two steps:</p><div class="orderedlist"><ol class="orderedlist" type="1"><li class="listitem"><p>The specification of the property to be verified by <a class="link" href="#iogma.gna.help.using.building_atomic_proposition" title="2.7.1.&nbsp;Specification of atomic propositions">defining
          atomic propositions</a> and <a class="link" href="#iogma.gna.help.using.formulating_temporal_logic_formulas" title="2.7.2.&nbsp;Formulation of temporal logic formulas">formulating
          so-called temporal-logic formulas</a>;</p></li><li class="listitem"><p>The actual verification of the property by a model
          checker.</p></li></ol></div><p>We will illustrate these steps by means of one of the
    above-mentioned properties for the example network.</p><div class="section" title="2.7.1.&nbsp;Specification of atomic propositions"><div class="titlepage"><div><div><h3 class="title"><a name="iogma.gna.help.using.building_atomic_proposition"></a>2.7.1.&nbsp;Specification of atomic propositions</h3></div></div></div><p>An <span class="bold"><strong>atomic proposition</strong></span> is used to
      characterize qualitative states. It indicates, for each variable in the
      model, its value, the value toward which it converges and the sign of
      its derivative. In addition, it specifies state descriptors indicating,
      <span class="emphasis"><em>e.g.</em></span>, whether the state is a steady state.</p><p>An atomic proposition can be created by means of <span class="guimenuitem">New
      atomic proposition</span> in the <span class="guilabel">Edit</span> menu:
      a new atomic proposition is created and the corresponding
      <span class="guilabel">Atomic proposition</span> window opens (see <a class="xref" href="#figure.atomic_proposition_editor" title="Fig.&nbsp;2.16.&nbsp;&nbsp;Atomic proposition windows, showing the specification of the atomic propositions InitialConditions and SteadyState.">Figure&nbsp;2.16</a>). The window is composed
      of 4 tabs:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p>the <span class="guilabel">Value</span> tab, where the user can choose
          (strict) lower and upper bounds for the value of each variable
          (following the same principle as the <a class="link" href="#iogma.gna.help.using.building_initial_conditions" title="2.3.&nbsp;Specifying initial conditions">specification
          of the initial conditions</a>);</p></li><li class="listitem"><p>the <span class="guilabel">FocalSet</span> tab, where the user can
          choose (strict) lower and upper bounds for the focal value of each
          variable. Within each region of the state space, the system
          converges towards a so-called focal set (see <a class="link" href="#iogma.gna.help.overview.whatis_gna" title="1.1.&nbsp;What is GNA?">the publications on
          GNA</a>);</p></li><li class="listitem"><p>the <span class="guilabel">Derivatives</span> tab, where for each
          variable, the user can select the appropriate derivative sign
          (<span class="guilabel">Undefined</span>, <span class="guilabel">Zero</span>,
          <span class="guilabel">Positive</span>, <span class="guilabel">Negative</span>,
          <span class="guilabel">All</span>);</p></li><li class="listitem"><p>the <span class="guilabel">StateDescriptors</span> tab contains check
          boxes that can be selected if the user wants to specify that the
          state is a steady state, an initial state, in a strongly connected
          component, or in a terminal cycle of the state transition
          graph.</p></li></ul></div><p>A newly-created atomic proposition has a default specification
      that does not impose any specific constraints, and is therefore
      <span class="bold"><strong>true</strong></span> for all states in the state
      transition graph.</p><p>In order to verify the property "<span class="emphasis"><em>If the system is in the
      state corresponding to the initial conditions <code class="varname">zero_a &lt;= x_a
      &lt; t_a1</code>, <code class="varname">zero_b &lt;= x_b &lt; t_b</code>, and
      <code class="varname">zero_c &lt;= x_c &lt; t_c1</code>, then it necessarily
      reaches a steady state</em></span>", we need to create two atomic
      propositions. One is called <code class="varname">InitialConditions</code> and
      specifies the initial conditions <code class="varname">zero_a &lt;= x_a &lt;
      t_a1</code>, <code class="varname">zero_b &lt;= x_b &lt; t_b</code>, and
      <code class="varname">zero_c &lt;= x_c &lt; t_c1</code>. The other is called
      <code class="varname">SteadyState</code> and specifies that the state is a steady
      state (see <a class="xref" href="#figure.atomic_proposition_editor" title="Fig.&nbsp;2.16.&nbsp;&nbsp;Atomic proposition windows, showing the specification of the atomic propositions InitialConditions and SteadyState.">Figure&nbsp;2.16</a>).</p><div class="figure"><a name="figure.atomic_proposition_editor"></a><p class="title">Fig.&nbsp;2.16.&nbsp;&nbsp;<span class="guilabel">Atomic proposition</span> windows, showing the
        specification of the atomic propositions
        <code class="varname">InitialConditions</code> and
        <code class="varname">SteadyState</code>.</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen13.png" align="middle" alt="Atomic proposition windows, showing the specification of the atomic propositions InitialConditions and SteadyState."></div><div class="mediaobject" align="center"><img src="images/GNAscreen13b.png" align="middle" alt="Atomic proposition windows, showing the specification of the atomic propositions InitialConditions and SteadyState."></div></div></div><br class="figure-break"></div><div class="section" title="2.7.2.&nbsp;Formulation of temporal logic formulas"><div class="titlepage"><div><div><h3 class="title"><a name="iogma.gna.help.using.formulating_temporal_logic_formulas"></a>2.7.2.&nbsp;Formulation of temporal logic formulas</h3></div></div></div><p>The atomic propositions are included in the temporal-logic
      property, defined by means of a special editor. The user can call the
      editor by opening the <span class="guimenu">Edit</span> menu and choosing the
      <span class="guimenuitem">New property</span> item. It opens the
      <span class="guilabel">Property editor</span> window which is composed of two
      tabs, corresponding to two different ways to formulate a property. The
      first is a pattern-based wizard that allows you to formulate a property
      without any knowledge of temporal logic. The second is a free-text
      interface that provides all the flexibility of the temporal logic
      supported by GNA, called CTRL. This logic subsumes standard temporal
      logics like CTL and LTL (see <a class="link" href="#iogma.gna.help.overview.whatis_gna" title="1.1.&nbsp;What is GNA?">GNA
      publications</a>).</p><p>A pattern is a high-level query template that captures recurring
      questions and can be automatically translated into temporal
      logic.</p><p>Four different patterns, each with some variants, are available to
      the user. Together they cover most of the frequently-asked questions by
      modelers (see <a class="link" href="#iogma.gna.help.overview.whatis_gna" title="1.1.&nbsp;What is GNA?">GNA
      publications</a>).</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p><span class="guilabel">Occurrence</span>: It is (not) possible for a
          state satisfying an atomic proposition <code class="varname">p</code> to
          occur;</p></li><li class="listitem"><p><span class="guilabel">Consequence</span>: If a state satisfying an
          atomic proposition <code class="varname">p</code> occurs, then it is possibly
          (or necessarily) followed by a state satisfying an atomic
          proposition <code class="varname">q</code>;</p></li><li class="listitem"><p><span class="guilabel">Sequence</span>: A state statisfying an atomic
          proposition <code class="varname">q</code> is reachable and is possibly (or
          necessarily) preceded at some time (or all of the time) by a state
          <code class="varname">p</code>;</p></li><li class="listitem"><p><span class="guilabel">Invariance</span>: A state satisfying an atomic
          proposition <code class="varname">p</code> can (or must) persist
          indefinitely.</p></li></ul></div><p>In the pattern wizard, you first click on the radio button for the
      pattern of your choice, and then use the combo boxes to choose existing
      atomic propositions and the specific pattern variant of interest (the
      default choice for the atomic propositions is
      <span class="guilabel">true</span>).</p><p>The temporal-logic translation of the property appears in the text
      area at the bottom of the window. This is illustrated by <a class="xref" href="#figure.property_editor_pattern" title="Fig.&nbsp;2.17.&nbsp;&nbsp;Pattern wizard of the property editor, showing the pattern definition of the property &quot;If the system is in the state corresponding to the initial conditions zero_a <= x_a < t_a1, zero_b <= x_b < t_b, and zero_c <= x_c < t_c1, then it necessarily reaches a steady state&quot; and its translation into temporal logic.">Figure&nbsp;2.17</a> for the property
      "<span class="emphasis"><em>If the system is in the state corresponding to the initial
      conditions <code class="varname">zero_a &lt;= x_a &lt; t_a1</code>,
      <code class="varname">zero_b &lt;= x_b &lt; t_b</code>, and <code class="varname">zero_c &lt;=
      x_c &lt; t_c1</code>, then it necessarily reaches a steady
      state</em></span>". The property is defined as a <span class="bold"><strong>Consequence</strong></span> pattern, using the previously-defined
      atomic propositions <code class="varname">InitialConditions</code> and
      <code class="varname">SteadyState</code>.</p><div class="figure"><a name="figure.property_editor_pattern"></a><p class="title">Fig.&nbsp;2.17.&nbsp;&nbsp;<span class="guilabel">Pattern</span> wizard of the property editor,
        showing the pattern definition of the property "<span class="emphasis"><em>If the
        system is in the state corresponding to the initial conditions
        <code class="varname">zero_a &lt;= x_a &lt; t_a1</code>, <code class="varname">zero_b &lt;=
        x_b &lt; t_b</code>, and <code class="varname">zero_c &lt;= x_c &lt;
        t_c1</code>, then it necessarily reaches a steady state</em></span>"
        and its translation into temporal logic.</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen14.png" align="middle" alt="Pattern wizard of the property editor, showing the pattern definition of the property &quot;If the system is in the state corresponding to the initial conditions zero_a <= x_a < t_a1, zero_b <= x_b < t_b, and zero_c <= x_c < t_c1, then it necessarily reaches a steady state&quot; and its translation into temporal logic."></div></div></div><br class="figure-break"><p>If you are comfortable with the CTRL temporal logic, you can
      alternatively select the <span class="guilabel">Expert</span> tab. This allows
      you to directly enter the property by typing it in the text area.
      Another way to edit a property is by using the combo box system,
      selecting the corresponding operators and atomic propositions, which are
      directly inserted in the text area, as shown in <a class="xref" href="#figure.property_editor_expert" title="Fig.&nbsp;2.18.&nbsp;&nbsp;Expert tab of the property editor, used for specifying the same property as in Figure&nbsp;2.17">Figure&nbsp;2.18</a>. To check if the temporal
      logic specification of the property is well-formed and compliant with
      the CTRL grammar, you can choose the <span class="guilabel">Check syntax</span>
      button. The syntax of CTRL formulas is defined <a class="ulink" href="iogma.gna.help.appendix.property_syntax" target="_top">in the
      appendix</a>.</p><p>The temporal-logic properties and atomic propositions are saved in
      the same project file of the model. However, they can also be imported
      and exported as files with the extension <code class="filename">.prop</code>. To this end, use the
      <span class="guimenuitem">Import</span> and <span class="guimenuitem">Export</span>
      items in the <span class="guimenu">File</span> menu.</p><div class="figure"><a name="figure.property_editor_expert"></a><p class="title">Fig.&nbsp;2.18.&nbsp;&nbsp;<span class="guilabel">Expert</span> tab of the property editor, used
        for specifying the same property as in <a class="xref" href="#figure.property_editor_pattern" title="Fig.&nbsp;2.17.&nbsp;&nbsp;Pattern wizard of the property editor, showing the pattern definition of the property &quot;If the system is in the state corresponding to the initial conditions zero_a <= x_a < t_a1, zero_b <= x_b < t_b, and zero_c <= x_c < t_c1, then it necessarily reaches a steady state&quot; and its translation into temporal logic.">Figure&nbsp;2.17</a></p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen15.png" align="middle" alt="Expert tab of the property editor, used for specifying the same property as in"></div></div></div><br class="figure-break"></div><div class="section" title="2.7.3.&nbsp;Verification of temporal logic formulas"><div class="titlepage"><div><div><h3 class="title"><a name="iogma.gna.help.using.ready_for_model_checking"></a>2.7.3.&nbsp;Verification of temporal logic formulas</h3></div></div></div><p>Once the temporal logic formula has been completely specified, you
      can check its validity by means of the model-checking server located at
      INRIA. This is done in two different ways: either through the property
      editor, by clicking on the <span class="guibutton">Specify verification</span>
      button, or through the item with the same name in the
      <span class="guilabel">Analysis</span> menu. In both cases, a
      <span class="guilabel">Verification</span> window is opened for the property to
      be verified. An example is shown in <a class="xref" href="#figure.go_for_model_checking" title="Fig.&nbsp;2.19.&nbsp;&nbsp;Verification window. Specification of the model checker options for the verification of the property defined in Figure&nbsp;2.17">Figure&nbsp;2.19</a>.</p><p>The <span class="guilabel">Verification</span> window asks you to specify
      the options of the model-checking process. The first option concerns the
      model checker. The choice of a model checker is constrained by the
      temporal logic used to specify the property and the available plugins.
      By default, the GNA distribution includes plugins for the NuSMV
      model-checker, which makes GNA capable of verifying properties specified
      in CTL. Plugins for other model-checkers are available on the <a class="ulink" href="http://ibis.inrialpes.fr/article122.html" target="_top">GNA
      website</a>.</p><p>After choosing a model-checker, the corresponding graph type is
      automatically selected. To each of the registered model checker is
      associated one of the following graph types:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p>The <span class="bold"><strong>explicit</strong></span> graph is an
          explicit export of the state transition graph previously obtained by
          running a simulation for a given set of initial conditions. If the
          graph is too big, it is not possible to use the model-checker web
          server;</p></li><li class="listitem"><p>The <span class="bold"><strong>implicit</strong></span> graph is a
          model-checker specific model description, containing all the
          necessary rules for the generation of the successors states by the
          model checker. This avoids having to run a simulation in GNA and
          ensures that a smaller file is sent through the web-server
          connection.</p></li></ul></div><p>If the <span class="bold"><strong>explicit</strong></span> graph is
      selected, you must choose one of the previously simulated state
      transition graphs. If the <span class="bold"><strong>implicit</strong></span>
      graph is selected, you must choose one of the previously defined initial
      conditions.</p><p>When clicking on the <span class="guilabel">Run</span> button, a
      verification request consisting of an (implicit or explicit) graph and a
      temporal-logic property is sent to the model-checker server. At any time
      you can abort the verification by clicking on the <span class="bold"><strong>Abort</strong></span> button.</p><p>Once the verification is completed, a result is returned by the
      model-checker server. This consists of two elements, displayed in the
      <span class="guilabel">Results</span> tab of the frame:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p>a verdict (true/false/error), representing the result of the
          property verification;</p></li><li class="listitem"><p>a diagnostic graph, containing an example or a counterexample
          (if one exists). In the explicit case, this counterexample will be a
          subgraph of the state transition graph sent to the model
          checker.</p></li></ul></div><p>If the verdict is <span class="bold"><strong>true</strong></span>, the
      diagnostic graph represents a path that shows the property to be
      correct. On the contrary, if the verdict is <span class="bold"><strong>false</strong></span>, the diagnostic graph represents a path
      that contradicts the property. Notice that some model checkers do not
      generate examples if the property is true (this is the case for
      NuSMV).</p><div class="figure"><a name="figure.go_for_model_checking"></a><p class="title">Fig.&nbsp;2.19.&nbsp;&nbsp;<span class="guilabel">Verification</span> window. Specification of the
        model checker options for the verification of the property defined in
        <a class="xref" href="#figure.property_editor_pattern" title="Fig.&nbsp;2.17.&nbsp;&nbsp;Pattern wizard of the property editor, showing the pattern definition of the property &quot;If the system is in the state corresponding to the initial conditions zero_a <= x_a < t_a1, zero_b <= x_b < t_b, and zero_c <= x_c < t_c1, then it necessarily reaches a steady state&quot; and its translation into temporal logic.">Figure&nbsp;2.17</a></p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen16.png" align="middle" alt="Verification window. Specification of the model checker options for the verification of the property defined in"></div></div></div><br class="figure-break"><p>In case you do not wish to use the model-checker server, or if the
      model-checker server is not available, you can also directly test your
      property in the model checker of your choice. To this end, install the
      model checker locally on your computer and export the (explicit or
      implicit) graph in the appropriate format. With the default plugin in
      the GNA distribution, the <span class="guimenu">File</span> menu has an
      <span class="guisubmenu">Export</span> item that allows you to generate files
      that can be read by NuSMV (<code class="filename">.smv</code>).
      The next distribution will also include the possibility to export an
      incomplete model files, without inequality constraints. You can also
      generate the graph from the console using the command explained in <a class="xref" href="#iogma.gna.help.using.running_simulations" title="2.4.&nbsp;Running simulations">Section&nbsp;2.4</a>. The supported
      formats are determined by the available plugins.</p><p>Examples of properties to be tested for several models in the GNA
      distribution can be consulted in the <code class="filename">samples</code> directory.</p></div></div><div class="section" title="2.8.&nbsp;Functions of the graphical user interface"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.using.gna_functions"></a>2.8.&nbsp;Functions of the graphical user interface</h2></div></div></div><p>In this section, the functions of GNA are summarized. The overview
    is structured according to the menus in the toolbar (<a class="xref" href="#figure.toolbar_network" title="Fig.&nbsp;2.20.&nbsp;&nbsp;Menu of the graphical user interface (network editor)">Figure&nbsp;2.20</a>, <a class="xref" href="#figure.toolbar_model" title="Fig.&nbsp;2.21.&nbsp;&nbsp;Menu of the graphical user interface (model editor)">Figure&nbsp;2.21</a>), which is slightly different depending
    on whether you are editing the network or the model.</p><div class="section" title="2.8.1.&nbsp;Network editor"><div class="titlepage"><div><div><h3 class="title"><a name="N109D6"></a>2.8.1.&nbsp;Network editor</h3></div></div></div><div class="figure"><a name="figure.toolbar_network"></a><p class="title">Fig.&nbsp;2.20.&nbsp;&nbsp;Menu of the graphical user interface (network editor)</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen11b.png" align="middle" alt="Menu of the graphical user interface (network editor)"></div></div></div><br class="figure-break"><div class="section" title="2.8.1.1.&nbsp;File menu"><div class="titlepage"><div><div><h4 class="title"><a name="N109E8"></a>2.8.1.1.&nbsp;<span class="guimenu">File</span> menu</h4></div></div></div><p>This menu contains functions to load and save files.</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">New project</span>: Start a new
            project;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">New model</span>: Start a new model
            (project without regulatory network);</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Open project...</span>: Open project
            stored in <code class="filename">.gnaml</code>
            file;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Open recent file</span>: Open a recently
            opened file (last 10 files are listed);</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Save project</span>: Save project to
            <code class="filename">.gnaml</code> file;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Save project as...</span>: Save project
            to <code class="filename">.gnaml</code> file under new
            name;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Import</span> </p><div class="itemizedlist"><ul class="itemizedlist" type="circle"><li class="listitem"><p><span class="guimenuitem">Import model</span>: Import model
                  stored in a <code class="filename">.gna</code>
                  file;</p></li><li class="listitem"><p><span class="guimenuitem">Import initial conditions</span>:
                  Import initial conditions stored in <code class="filename">.dat</code> file;</p></li><li class="listitem"><p><span class="guimenuitem">Import property...</span>: Import
                  property stored in <code class="filename">.prop</code> file;</p></li><li class="listitem"><p><span class="guimenuitem">Import from SBML</span>: Import
                  model and initial conditions from an SBML Level 2
                  document;</p></li></ul></div></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Export</span> </p><div class="itemizedlist"><ul class="itemizedlist" type="circle"><li class="listitem"><p><span class="guimenuitem">Export model...</span>: Export
                  model to <code class="filename">.gna</code>
                  file;</p></li><li class="listitem"><p><span class="guimenuitem">Export initial
                  conditions...</span>: Export initial conditions to
                  <code class="filename">.dat</code> file;</p></li><li class="listitem"><p><span class="guimenuitem">Export property...</span>: Export
                  property to <code class="filename">.prop</code>
                  file;</p></li><li class="listitem"><p><span class="guimenuitem">Export to SBML</span>: Export
                  model and initial conditions to an SBML document;</p></li></ul></div></li></ul></div><p>Some of the above import and export functions are accessible by
        clicking the <span class="mousebutton">right</span> mouse button in the
        <span class="guilabel">Project</span> tree or in the corresponding
        windows.</p></div><div class="section" title="2.8.1.2.&nbsp;Edit menu"><div class="titlepage"><div><div><h4 class="title"><a name="N10A8C"></a>2.8.1.2.&nbsp;Edit menu</h4></div></div></div><p>This menu contains functions for editing the
        network.</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span class="guisubmenu">New</span> </p><div class="itemizedlist"><ul class="itemizedlist" type="circle"><li class="listitem"><p><span class="guimenuitem">Gene</span>: Create a new
                    gene;</p></li><li class="listitem"><p><span class="guimenuitem">Regulating entity</span>: Create
                    new regulating entity;</p></li><li class="listitem"><p><span class="guimenuitem">Degradation</span>: Create new
                    degradation product for an entity;</p></li><li class="listitem"><p><span class="guimenuitem">Complex</span>: Create new
                    complex made of two or more entities;</p></li><li class="listitem"><p><span class="guimenuitem">Conversion</span>: Create
                    conversion link between two or more entities;</p></li><li class="listitem"><p><span class="guimenuitem">Transport</span>: Create
                    transport link between two entities;</p></li><li class="listitem"><p><span class="guimenuitem">Activation</span>: Create
                    activation link between an entity and a reaction;</p></li><li class="listitem"><p><span class="guimenuitem">Inhibition</span>: Create
                    inhibition link between an entity and a reaction;</p></li><li class="listitem"><p><span class="guimenuitem">Catalysis</span>: Create
                    catalysis link between an entity and a reaction.</p></li></ul></div></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Rename</span>: Rename selected
              entity;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Switch gene orientation</span>: Switch
              gene orientation from right to left to left to right, or
              reverse;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Add promoter</span>: Add new promoter
              to selected gene;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Delete</span>: Delete selected entity,
              reaction or regulation;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Preferences</span>: Edit preferences
              for GNA.</p></li></ul></div><p>Most of the above functions are also accessible by clicking the
        <span class="mousebutton">right</span> mouse button in the editor window or
        by means of the toolbar.</p></div><div class="section" title="2.8.1.3.&nbsp;View menu"><div class="titlepage"><div><div><h4 class="title"><a name="N10AF5"></a>2.8.1.3.&nbsp;View menu</h4></div></div></div><p>This menu contains functions to view and analyze the
        network.</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p><span class="guimenuitem">Show whole network</span>: Display
              whole network as opposed to the reduced network;</p></li><li class="listitem"><p><span class="guimenuitem">Show reduced network</span>: Display
              reduced network;</p></li><li class="listitem"><p><span class="guimenuitem">Zoom in</span>: Zoom in on
              network;</p></li><li class="listitem"><p><span class="guimenuitem">Zoom out</span>: Zoom out of
              network.</p></li><li class="listitem"><p><span class="guimenuitem">Fit content</span>: Zoom in or out so
              that the whole network is displayed in the window</p></li><li class="listitem"><p><span class="guimenuitem">Reset zoom</span>: Zoom to 1:1
              level</p></li></ul></div></div><div class="section" title="2.8.1.4.&nbsp;Help menu"><div class="titlepage"><div><div><h4 class="title"><a name="N10B1F"></a>2.8.1.4.&nbsp;<span class="guimenu">Help</span> menu</h4></div></div></div><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Help</span>: Open help pages;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">About GNA</span>: Display GNA
            version.</p></li></ul></div></div></div><div class="section" title="2.8.2.&nbsp;Model editor"><div class="titlepage"><div><div><h3 class="title"><a name="N10B34"></a>2.8.2.&nbsp;Model editor</h3></div></div></div><div class="figure"><a name="figure.toolbar_model"></a><p class="title">Fig.&nbsp;2.21.&nbsp;&nbsp;Menu of the graphical user interface (model editor)</p><div class="figure-contents"><div class="mediaobject" align="center"><img src="images/GNAscreen11.png" align="middle" alt="Menu of the graphical user interface (model editor)"></div></div></div><br class="figure-break"><div class="section" title="2.8.2.1.&nbsp;File menu"><div class="titlepage"><div><div><h4 class="title"><a name="N10B46"></a>2.8.2.1.&nbsp;<span class="guimenu">File</span> menu</h4></div></div></div><p>This menu contains functions to load and save files.</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">New project</span>: Start a new
            project;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">New model</span>: Start a new model
            (project without regulatory network);</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Open project...</span>: Open project
            stored in <code class="filename">.gnaml</code>
            file;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Open recent file</span>: Open a recently
            opened file (last 10 files are listed);</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Save project</span>: Save project to
            <code class="filename">.gnaml</code> file;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Save project as...</span>: Save project
            to <code class="filename">.gnaml</code> file under new
            name;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Import</span> </p><div class="itemizedlist"><ul class="itemizedlist" type="circle"><li class="listitem"><p><span class="guimenuitem">Import model</span>: Import model
                  stored in a <code class="filename">.gna</code>
                  file;</p></li><li class="listitem"><p><span class="guimenuitem">Import initial conditions</span>:
                  Import initial conditions stored in <code class="filename">.dat</code> file;</p></li><li class="listitem"><p><span class="guimenuitem">Import property...</span>: Import
                  property stored in <code class="filename">.prop</code> file;</p></li><li class="listitem"><p><span class="guimenuitem">Import from SBML</span>: Import
                  model and initial conditions from an SBML Level 2
                  document;</p></li></ul></div></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Export</span> </p><div class="itemizedlist"><ul class="itemizedlist" type="circle"><li class="listitem"><p><span class="guimenuitem">Export model...</span>: Export
                  model to <code class="filename">.gna</code>
                  file;</p></li><li class="listitem"><p><span class="guimenuitem">Export initial
                  conditions...</span>: Export initial conditions to
                  <code class="filename">.dat</code> file;</p></li><li class="listitem"><p><span class="guimenuitem">Export property...</span>: Export
                  property to <code class="filename">.prop</code>
                  file;</p></li><li class="listitem"><p><span class="guimenuitem">Export to SBML</span>: Export
                  model and initial conditions to an SBML document;</p></li><li class="listitem"><p><span class="guimenuitem">Export explicit graph to &lt;model
                  checker&gt;...</span>: Export state transition graph
                  in a file that can be read by the &lt;model
                  checker&gt;;</p></li><li class="listitem"><p><span class="guimenuitem">Export implicit graph to &lt;model
                  checker&gt;...</span>: Export implicit graph defined
                  by given initial conditions in a file that can be read by
                  the &lt;model checker&gt;;</p></li></ul></div></li></ul></div><p>Some of the above import and export functions are accessible by
        clicking the <span class="mousebutton">right</span> mouse button in the
        <span class="guilabel">Project</span> tree or in the corresponding
        windows.</p></div><div class="section" title="2.8.2.2.&nbsp;Edit menu"><div class="titlepage"><div><div><h4 class="title"><a name="N10BF6"></a>2.8.2.2.&nbsp;<span class="guimenu">Edit</span> menu</h4></div></div></div><p>This menu contains functions for editing the model, in
        particular the variables composing the model.</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span class="guisubmenu">New</span> </p><div class="itemizedlist"><ul class="itemizedlist" type="circle"><li class="listitem"><p><span class="guimenuitem">Variable</span>: Create a new
                  variable;</p></li><li class="listitem"><p><span class="guimenuitem">Initial conditions</span>: Create
                  new initial conditions;</p></li><li class="listitem"><p><span class="guimenuitem">Atomic proposition</span>: Create
                  new atomic proposition;</p></li><li class="listitem"><p><span class="guimenuitem">Property</span>: Create new
                  property;</p></li></ul></div></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Edit</span>: Edit selected variable,
            initial conditions, atomic proposition or property;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Rename</span>: Rename selected variable,
            initial conditions, atomic proposition or property;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Delete</span>: Delete selected variable,
            initial conditions, atomic proposition or property;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">View comment</span>: View comment for
            selected variable, initial conditions, atomic proposition or
            property;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Preferences</span>: Edit preferences for
            GNA.</p></li></ul></div><p>Most of the above functions are also accessible by clicking the
        <span class="mousebutton">right</span> mouse button in the
        <span class="guilabel">Project</span> window. The <span class="guilabel">Variable
        window</span> for a variable contains an equation editor for
        defining the state equation. The following key combinations can be
        used to edit the information in the <span class="guilabel">State
        equation</span> field:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span class="keycap"><strong>Ctrl</strong></span>+<span class="keycap"><strong>X</strong></span>: Cut and add to buffer;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="keycap"><strong>Ctrl</strong></span>+<span class="keycap"><strong>C</strong></span>: Copy and add to buffer;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="keycap"><strong>Ctrl</strong></span>+<span class="keycap"><strong>V</strong></span>: Paste buffer;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="keycap"><strong>Ctrl</strong></span>+<span class="keycap"><strong>Z</strong></span>: Undo;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="keycap"><strong>Ctrl</strong></span>+<span class="keycap"><strong>Y</strong></span>: Redo;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="keycap"><strong>Ctrl</strong></span>+<span class="keycap"><strong>space</strong></span>: Complete current name.</p></li></ul></div></div><div class="section" title="2.8.2.3.&nbsp;Analysis menu"><div class="titlepage"><div><div><h4 class="title"><a name="N10C9C"></a>2.8.2.3.&nbsp;<span class="guimenu">Analysis</span> menu</h4></div></div></div><p>This menu contains functions to analyze the model
        defined.</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Search attractors</span>: Search phase
            space for qualitative steady states of the model;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Specify simulation</span>: Specify
            simulation parameters for given initial conditions;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Specify verification</span>: Specify
            verfication parameters for given property.</p></li></ul></div><p>The above functions are also accessible by clicking the
        <span class="mousebutton">right</span> mouse button in the
        <span class="guilabel">Project</span> tree.</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p><span class="guimenuitem">Run</span>: Run specified attractor
            search, simulation or verification;</p></li><li class="listitem"><p><span class="guimenuitem">Abort</span>: Abort current running
            attractor search, simulation or verification.</p></li></ul></div><p>These function are also accessible from the buttons in each
        <span class="guilabel">Attractor search</span>, <span class="guilabel">Simulation</span>
        or <span class="guilabel">Verification</span> window.</p></div><div class="section" title="2.8.2.4.&nbsp;Results menu (model editor mode)"><div class="titlepage"><div><div><h4 class="title"><a name="N10CDE"></a>2.8.2.4.&nbsp;<span class="guimenu">Results</span> menu (model editor mode)</h4></div></div></div><p>This menu contains functions to view and analyze simulation
        results.</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Deselect all states</span>: Deselect all
            states in the state transition graph;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Filter instantaneous states</span>:
            Generate the state transition graph from which all states that are
            instantaneously traversed have been removed. Display this graph in
            a new tab;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Subgraph spanning selected
            states</span>: Construct the subgraph spanning the selected
            states in the state transition graph. Display this graph in a new
            tab;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Expand selected subgraph</span>: Expand
            the state transition graph using several levels of successors
            and/or predecessors of the selected states. Display this graph in
            a new tab;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Display variables in selected
            path</span>: Display the qualitative evolution of the
            variables in a selected path in the state transition graph;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Display focal sets in selected
            path</span>: Display the qualitative evolution of the
            values toward which the state variables locally converge;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Zoom in graph</span>: Zoom in on state
            transition graph;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Zoom out graph</span>: Zoom out of state
            transition graph.</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Fit graph</span>: Zoom in or out so that
            the whole state transition graph is displayed in the window</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Reset zoom</span>: Zoom to 1:1
            level</p></li></ul></div><p>The above functions are also accessible by clicking the
        <span class="mousebutton">right</span> mouse button in the
        <span class="guilabel">Transition graph</span> window. States in the
        <span class="guilabel">Transition graph</span> window can be selected by means
        of the mouse while pressing the <span class="keycap"><strong>Ctrl</strong></span> key. Deselection is achieved by
        clicking and pressing the <span class="keycap"><strong>Ctrl</strong></span>
        key. Some shortcuts allow to easily navigate within the graph: while
        maintaining the <span class="keycap"><strong>Ctrl</strong></span> and <span class="keycap"><strong>Alt</strong></span> keys pressed, you can drag the graph with
        the mouse ; when the <span class="keycap"><strong>Ctrl</strong></span> key is
        pressed, the mouse wheel allows to zoom in and out.</p></div><div class="section" title="2.8.2.5.&nbsp;Windows menu (model editor mode)"><div class="titlepage"><div><div><h4 class="title"><a name="N10D4C"></a>2.8.2.5.&nbsp;<span class="guimenu">Windows</span> menu (model editor mode)</h4></div></div></div><p>This menu lists the active windows and allows you to make some
        of them invisible.</p></div><div class="section" title="2.8.2.6.&nbsp;Help menu"><div class="titlepage"><div><div><h4 class="title"><a name="N10D54"></a>2.8.2.6.&nbsp;<span class="guimenu">Help</span> menu</h4></div></div></div><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">Help</span>: Open help pages;</p></li><li class="listitem" style="list-style-type: disc"><p><span class="guimenuitem">About GNA</span>: Display GNA
            version.</p></li></ul></div></div></div></div></div><div class="chapter" title="Part 3. Appendices"><div class="titlepage"><div><div><h2 class="title"><a name="N10D69"></a>Part 3. Appendices</h2></div></div></div><div class="section" title="3.1.&nbsp;Syntax of GNA model"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.appendix.model_syntax"></a>3.1.&nbsp;Syntax of GNA model</h2></div></div></div><p>Model files have the extension <code class="filename">.gna</code> and can be
    edited using any text editor.</p><div class="section" title="3.1.1.&nbsp;Syntactic conventions"><div class="titlepage"><div><div><h3 class="title"><a name="N10D77"></a>3.1.1.&nbsp;Syntactic conventions</h3></div></div></div><p>Terminals are written in <span class="bold"><strong>bold</strong></span>,
      whereas non-terminals are written in <span class="emphasis"><em>italics</em></span>.
      Furthermore,</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p>| &nbsp;separates alternatives;</p></li><li class="listitem" style="list-style-type: disc"><p>(...) delimitates a block composed of several
          alternatives;</p></li><li class="listitem" style="list-style-type: disc"><p>[...] delimitates an optional block;</p></li><li class="listitem" style="list-style-type: disc"><p>[...]<sup>*</sup> denotes a block repeated any
          (possibly 0) number of times;</p></li><li class="listitem" style="list-style-type: disc"><p>[...]<sup>+</sup> denotes a block repeated at
          least once.</p></li></ul></div></div><div class="section" title="3.1.2.&nbsp;BNF of a model"><div class="titlepage"><div><div><h3 class="title"><a name="N10D9B"></a>3.1.2.&nbsp;BNF of a model</h3></div></div></div><p>The BNF syntax of the GNA models is given below. The axiom is
      <span class="emphasis"><em>model</em></span>.</p><div class="sidebar"><p class="title"></p><p><span class="emphasis"><em>model</em></span> ::=
        [<span class="emphasis"><em>variable</em></span>]<sup>+</sup></p><p><span class="emphasis"><em>variable</em></span> ::=<span class="bold"><strong>
        state-variable:</strong></span><span class="emphasis"><em> idf
        </em></span>[<span class="emphasis"><em>variable-description</em></span>]<sup>+</sup>
        | <span class="bold"><strong>input-variable:</strong></span><span class="emphasis"><em> idf
        </em></span>[<span class="emphasis"><em>variable-description</em></span>]<sup>+</sup></p><p><span class="emphasis"><em>variable-description</em></span> ::=</p><p><span class="bold"><strong>synthesis-parameters:</strong></span>&nbsp;
        <span class="emphasis"><em>idf </em></span>[<span class="emphasis"><em><span class="bold"><strong>,</strong></span>
        idf</em></span>]<sup>*</sup> | <span class="bold"><strong>degradation-parameters:</strong></span>&nbsp;
        <span class="emphasis"><em>idf</em></span> [<span class="emphasis"><em><span class="bold"><strong>,</strong></span>
        idf</em></span>]<sup>*</sup> | <span class="bold"><strong>threshold-parameters:&nbsp;</strong></span> <span class="emphasis"><em>idf
        </em></span>[<span class="emphasis"><em><span class="bold"><strong>,</strong></span>
        idf</em></span>]<sup>*</sup><span class="emphasis"><em> </em></span>|
        <span class="bold"><strong>box-parameter:&nbsp;</strong></span> <span class="emphasis"><em>idf
        </em></span> | <span class="bold"><strong>zero-parameter:&nbsp;</strong></span><span class="emphasis"><em>idf</em></span> |
        <span class="bold"><strong>state-equation:&nbsp;</strong></span>
        <span class="emphasis"><em>state-equation </em></span>| <span class="bold"><strong>threshold-inequalities:</strong></span><span class="emphasis"><em>
        </em></span>[<span class="emphasis"><em>threshold-inequality</em></span><span class="bold"><strong> ;</strong></span> ]<sup>+</sup> | <span class="bold"><strong>equilibrium-inequalities:&nbsp;</strong></span>
        [<span class="emphasis"><em>equilibrium-inequality</em></span><span class="bold"><strong>
        ;</strong></span> ]<sup>+</sup></p><p><span class="emphasis"><em>state-equation</em></span> ::= <span class="bold"><strong>d/dt</strong></span> <span class="emphasis"><em>idf</em></span> <span class="bold"><strong>= </strong></span><span class="emphasis"><em>synthesis-term </em></span><span class="bold"><strong>-</strong></span> <span class="emphasis"><em>degradation-term</em></span></p><p><span class="emphasis"><em>synthesis-term</em></span> ::= <span class="emphasis"><em>idf</em></span>
        | [<span class="emphasis"><em>idf</em></span><span class="bold"><strong>+</strong></span> ]
        <span class="emphasis"><em>idf</em></span> <span class="bold"><strong>*</strong></span>
        <span class="emphasis"><em>step-expr</em></span> [<span class="bold"><strong>+</strong></span>
        <span class="emphasis"><em>idf</em></span><span class="bold"><strong>*</strong></span>
        <span class="emphasis"><em>step-expr</em></span>]<sup>*</sup></p><p><span class="emphasis"><em>degradation-term</em></span> ::=
        <span class="emphasis"><em>idf</em></span> <span class="bold"><strong>*</strong></span>
        <span class="emphasis"><em>idf</em></span> | <span class="bold"><strong>(</strong></span>
        <span class="emphasis"><em>idf</em></span>&nbsp;[<span class="bold"><strong>+</strong></span>
        <span class="emphasis"><em>idf</em></span> <span class="bold"><strong>*</strong></span>
        <span class="emphasis"><em>step-expr</em></span>]<sup>+</sup> <span class="bold"><strong>)</strong></span> <span class="bold"><strong>*</strong></span>
        <span class="emphasis"><em>idf</em></span></p><p><span class="emphasis"><em>step-expr</em></span> ::= <span class="bold"><strong>( 1 -
        </strong></span><span class="emphasis"><em>step-expr</em></span> <span class="bold"><strong>)</strong></span> | <span class="emphasis"><em>step-expr</em></span> <span class="bold"><strong>*</strong></span> <span class="emphasis"><em>step-expr</em></span> | <span class="bold"><strong>s+( </strong></span><span class="emphasis"><em>idf<span class="bold"><strong>
        ,</strong></span> idf</em></span><span class="bold"><strong> ) | s-(
        </strong></span><span class="emphasis"><em>idf<span class="bold"><strong> ,</strong></span>
        idf</em></span><span class="bold"><strong> )</strong></span></p><p><span class="emphasis"><em>inequality-term ::= idf |
        focal-param-expression</em></span></p><p><span class="emphasis"><em>focal-param-expression ::= synthesis-param-sum
        <span class="bold"><strong>/</strong></span>
        degradation-param-sum</em></span></p><p><span class="emphasis"><em>synthesis-param-sum ::= idf | <span class="bold"><strong>(</strong></span> idf [<span class="bold"><strong>+</strong></span>
        idf]<sup>+</sup> <span class="bold"><strong>)</strong></span>
        </em></span></p><p><span class="emphasis"><em>degradation-param-sum ::= idf | <span class="bold"><strong>(</strong></span> idf [<span class="bold"><strong>+</strong></span>
        idf]<sup>+</sup> <span class="bold"><strong>)</strong></span></em></span></p><p><span class="emphasis"><em>idf</em></span> ::= (<span class="bold"><strong>a</strong></span>
        |..| <span class="bold"><strong>z </strong></span>|<span class="bold"><strong>
        A</strong></span> |..| <span class="bold"><strong>Z</strong></span>) [<span class="bold"><strong>a</strong></span> |..| <span class="bold"><strong>z
        </strong></span>|<span class="bold"><strong> A</strong></span> |..| <span class="bold"><strong>Z </strong></span>|<span class="bold"><strong> 0</strong></span> |..|
        <span class="bold"><strong>9 </strong></span>|<span class="bold"><strong>
        _</strong></span> ]<sup>*</sup></p></div><p>Comments can be inserted into GNA models. The syntax of comments
      follows the rules for comments in languages like C: multi-line comments
      are placed between <span class="bold"><strong>/*</strong></span> and <span class="bold"><strong>*/</strong></span>, while single-line comments are placed after
      characters <span class="bold"><strong>//</strong></span>.</p></div><div class="section" title="3.1.3.&nbsp;Additional constraints"><div class="titlepage"><div><div><h3 class="title"><a name="N10F34"></a>3.1.3.&nbsp;Additional constraints</h3></div></div></div><p>In addition to the above syntax, a number of constraints must be
      respected:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p>Some keywords are reserved and cannot be used as identifiers:
          <span class="bold"><strong>true</strong></span> and <span class="bold"><strong>false</strong></span>. In addition to this, it is recommended
          not to use the following identifiers if you intend to use GNA's
          model checking tools: <span class="bold"><strong>equ</strong></span>,
          <span class="bold"><strong>implies</strong></span>, <span class="bold"><strong>or</strong></span>, <span class="bold"><strong>and</strong></span>,
          <span class="bold"><strong>not</strong></span>, <span class="bold"><strong>ef</strong></span>, <span class="bold"><strong>af</strong></span>,
          <span class="bold"><strong>eg</strong></span>, <span class="bold"><strong>ag</strong></span>, <span class="bold"><strong>ex</strong></span>,
          <span class="bold"><strong>ax</strong></span>, <span class="bold"><strong>nil</strong></span>, <span class="bold"><strong>e</strong></span>,
          <span class="bold"><strong>a</strong></span>, <span class="bold"><strong>u</strong></span> and <span class="bold"><strong>r</strong></span>.</p></li><li class="listitem" style="list-style-type: disc"><p>If the variable is an input variable, only the
          threshold-parameters, box-parameter, zero-parameter, and
          threshold-inequalities blocks can appear in a variable block.</p></li><li class="listitem" style="list-style-type: disc"><p>The synthesis-parameters block must appear exactly once in a
          variable block. Each synthesis parameter used in the right-hand side
          of the state equation in some variable block must be declared in the
          synthesis-parameters block. In previous versions of GNA, the
          terminal <span class="bold"><strong>production-parameters:</strong></span> was
          used instead of <span class="bold"><strong>synthesis-parameters:</strong></span>. This alternative
          terminal is still accepted by the current version.</p></li><li class="listitem" style="list-style-type: disc"><p>The degradation-parameters block must appear exactly once in a
          variable block.</p></li><li class="listitem" style="list-style-type: disc"><p>The threshold-parameters block must appear at most once in a
          variable block. (It can be omitted if the state or input variable
          has no associated threshold parameters.) The threshold parameters in
          the thresholds-parameters block of a variable <code class="varname">x</code>
          are the thresholds at which <code class="varname">x</code> regulates other
          variables. That is, the threshold parameters may appear in step
          functions in other variable blocks.</p></li><li class="listitem" style="list-style-type: disc"><p>The zero-parameter and box-parameter blocks must appear
          exactly once in a variable block.</p></li><li class="listitem" style="list-style-type: disc"><p>The state-equation block must appear exactly once in a
          variable block. The <span class="emphasis"><em>idf</em></span> in the left-hand side
          of the state equation must be the variable being described.</p></li><li class="listitem" style="list-style-type: disc"><p>The synthesis-parameters, degradation-parameters,
          threshold-parameters, zero-parameter and maximum-parameter blocks
          must have been defined before the state equation and
          parameter-inequalities blocks.</p></li><li class="listitem" style="list-style-type: disc"><p>In the synthesis term of the state equation, each
          <span class="emphasis"><em>idf</em></span> corresponds to a synthesis parameter of the
          variable being described.</p></li><li class="listitem" style="list-style-type: disc"><p>In the degradation term of the state equation, the rightmost
          <span class="emphasis"><em>idf</em></span> in each alternative corresponds to the
          variable being described, while the others correspond to degradation
          parameters of that variable.</p></li><li class="listitem" style="list-style-type: disc"><p>In step-function expressions, the first
          <span class="emphasis"><em>idf</em></span> is the regulator variable and the second
          <span class="emphasis"><em>idf</em></span> a threshold parameter of this
          variable.</p></li><li class="listitem" style="list-style-type: disc"><p>The parameter-inequalities block must appear exactly once in a
          variable block. In order to make simulation possible, these
          inequalities must induce a total order on the threshold, focal,
          zero, and maximum parameters of the variable. The focal parameters
          occurring in the parameter inequality are determined from the state
          equation by taking all possible quotients of synthesis and
          degradation parameters, except 0. For instance, consider the
          following state equation: <code class="varname">d/dt x_a = k_a1 + k_a2 *
          s+(x_b,t_b) - g_a * x_a</code>.The possible focal parameters are
          in this case <code class="varname">k_a1/ g_a</code> and <code class="varname">(k_a1+ k_a2)/
          g_a</code>.The user has to assure that the set of inequality
          constraints specified is consistent. Notice that the terminals
          <span class="bold"><strong>threshold-inequalities:</strong></span> and
          <span class="bold"><strong>equilibrium-inequalities:</strong></span>
          (<span class="bold"><strong>nullcline-inequalities:</strong></span>) are no
          longer used in GNA 6.0.</p></li><li class="listitem" style="list-style-type: disc"><p>The first inequality term in a parameter equality must be the
          zero parameter, while the last inequality term must be the maximum
          parameter.</p></li></ul></div><p>Examples of model files can be consulted in the
      <code class="filename">samples</code> directory of the GNA distribution.</p></div></div><div class="section" title="3.2.&nbsp;Syntax of GNA initial conditions"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.appendix.initialconditions_syntax"></a>3.2.&nbsp;Syntax of GNA initial conditions</h2></div></div></div><p>The initial conditions file specifies the initial domain(s), and
    hence initial qualitative state(s), from which the genetic regulatory
    network is to be simulated. Initial conditions files have the
    extension<code class="filename">.dat</code> and can be edited using any text
    editor.</p><div class="section" title="3.2.1.&nbsp;Syntactic conventions"><div class="titlepage"><div><div><h3 class="title"><a name="N10FED"></a>3.2.1.&nbsp;Syntactic conventions</h3></div></div></div><p>Terminals are written in <span class="bold"><strong>bold</strong></span>,
      whereas non-terminals are written in <span class="emphasis"><em>italics</em></span>.
      Furthermore,</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p>| separates alternatives;</p></li><li class="listitem" style="list-style-type: disc"><p>(...) delimitates a block composed of several
          alternatives;</p></li><li class="listitem" style="list-style-type: disc"><p>[...] delimitates an optional block;</p></li><li class="listitem" style="list-style-type: disc"><p>[...]<sup>*</sup> denotes a block repeated any
          (possibly 0) number of times;</p></li><li class="listitem" style="list-style-type: disc"><p>[...]<sup>+ </sup>denotes a block repeated at
          least once.</p></li></ul></div></div><div class="section" title="3.2.2.&nbsp;BNF of an initial conditions file"><div class="titlepage"><div><div><h3 class="title"><a name="N11011"></a>3.2.2.&nbsp;BNF of an initial conditions file</h3></div></div></div><p>The BNF definition of the syntax of the GNA initial conditions is
      shown below. The axiom is
      <span class="emphasis"><em>initial-conditions</em></span>.</p><div class="sidebar"><p class="title"></p><p><span class="emphasis"><em>initial-conditions</em></span> ::= <span class="bold"><strong>initial-conditions:
        </strong></span>[<span class="emphasis"><em>initial-condition</em></span> <span class="bold"><strong>;</strong></span> ]<sup>+</sup></p><p><span class="emphasis"><em>initial-condition </em></span>::=
        <span class="emphasis"><em>idf</em></span> <span class="bold"><strong>&lt;</strong></span><span class="emphasis"><em> idf</em></span> |
        <span class="emphasis"><em>idf</em></span> <span class="bold"><strong>&gt;</strong></span><span class="emphasis"><em> idf</em></span>
        |<span class="emphasis"><em>idf</em></span> <span class="bold"><strong>&lt;=</strong></span><span class="emphasis"><em> idf</em></span>
        |<span class="emphasis"><em>idf</em></span> <span class="bold"><strong>&gt;=</strong></span><span class="emphasis"><em> idf</em></span></p><p><span class="emphasis"><em>idf</em></span> ::= (<span class="bold"><strong>a</strong></span>
        |..| <span class="bold"><strong>z </strong></span>|<span class="bold"><strong>
        A</strong></span> |..| <span class="bold"><strong>Z</strong></span>) [<span class="bold"><strong>a</strong></span> |..|<span class="bold"><strong>z
        </strong></span>|<span class="bold"><strong> A</strong></span> |..| <span class="bold"><strong>Z </strong></span>|<span class="bold"><strong> 0</strong></span> |..|
        <span class="bold"><strong>9 </strong></span>|<span class="bold"><strong> _
        </strong></span>]*</p></div><p>Comments can be inserted into the files. The syntax of comments
      follows the rules for comments in languages like C: multi-line comments
      are placed between <span class="bold"><strong>/*</strong></span> and <span class="bold"><strong>*/</strong></span>, while single-line comments are placed after
      characters <span class="bold"><strong>//</strong></span>.</p></div><div class="section" title="3.2.3.&nbsp;Additional constraints"><div class="titlepage"><div><div><h3 class="title"><a name="N11090"></a>3.2.3.&nbsp;Additional constraints</h3></div></div></div><p>In the above syntax, for each of the alternative
      <span class="emphasis"><em>initial-condition</em></span> expressions, the first
      <span class="emphasis"><em>idf</em></span> is a state or input variable and the second
      <span class="emphasis"><em>idf</em></span> either the box parameter, the zero parameter,
      or one of the threshold parameters of the variable. Initial conditions
      come in pairs for each variable, the first one giving its lower bound,
      the second one its upper bound. For the lower bound, the operator must
      be <span class="bold"><strong>&gt;=</strong></span>, if the right-hand side
      parameter is a zero parameter, and <span class="bold"><strong>&gt;</strong></span>, if it is a threshold parameter. For the
      upper bound, the operator must be <span class="bold"><strong>&lt;=</strong></span>, if the right-hand side parameter is a box
      parameter, and <span class="bold"><strong>&lt;</strong></span>, if it is a
      threshold parameter. The order of the threshold parameters, zero
      parameters, and box parameters given in the model file must be
      respected. Examples of initial conditions files can be consulted in the
      <code class="filename">samples</code> directory of the GNA distribution.</p></div></div><div class="section" title="3.3.&nbsp;Syntax of CTRL property"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="iogma.gna.help.appendix.property_syntax"></a>3.3.&nbsp;Syntax of CTRL property</h2></div></div></div><p>The property file specifies a biological property enconded with a
    CTRL temporal logic formula, and the corresponding Atomic Propositions
    (describing restrictions on a given state) that are referred to inside
    this formula. Property files have the extension <code class="filename">.prop</code> and can be edited using any text
    editor.</p><div class="section" title="3.3.1.&nbsp;Syntactic conventions"><div class="titlepage"><div><div><h3 class="title"><a name="N110BD"></a>3.3.1.&nbsp;Syntactic conventions</h3></div></div></div><p>Terminals are written in <span class="bold"><strong>bold</strong></span>,
      whereas non-terminals are written in <span class="emphasis"><em>italics</em></span>.
      Furthermore,</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p>|&nbsp;separates alternatives;</p></li><li class="listitem" style="list-style-type: disc"><p>(...) delimitates a block composed of several
          alternatives;</p></li><li class="listitem" style="list-style-type: disc"><p>[...] delimitates an optional block;</p></li><li class="listitem" style="list-style-type: disc"><p>[...]<sup>*</sup> denotes a block repeated any
          (possibly 0) number of times;</p></li><li class="listitem" style="list-style-type: disc"><p>[...]<sup>+</sup> denotes a block repeated at
          least once.</p></li></ul></div></div><div class="section" title="3.3.2.&nbsp;BNF of a property"><div class="titlepage"><div><div><h3 class="title"><a name="N110E1"></a>3.3.2.&nbsp;BNF of a property</h3></div></div></div><p>The BNF syntax of a property is given below. The axiom is
      <span class="emphasis"><em>property</em></span>.</p><div class="sidebar"><p class="title"></p><p><span class="emphasis"><em>property</em></span> ::=
        [<span class="emphasis"><em>atomic-propositions</em></span>]<sup>*</sup>
        <span class="bold"><strong>property:</strong></span> <span class="emphasis"><em>idf </em></span>
        <span class="bold"><strong>formula:</strong></span> <span class="emphasis"><em>property-formula
        </em></span><span class="bold"><strong>;</strong></span></p><p><span class="emphasis"><em>atomic-propositions</em></span> ::= <span class="bold"><strong>atomic-proposition:</strong></span><span class="emphasis"><em> idf
        </em></span>[<span class="emphasis"><em>subatomic-proposition</em></span>]<sup>+</sup></p><p><span class="emphasis"><em>subatomic-proposition</em></span> ::=</p><p><span class="bold"><strong> domain:</strong></span>
        [<span class="emphasis"><em>idf</em></span> <span class="emphasis"><em>inequality-term</em></span>
        <span class="emphasis"><em>boundary-parameter</em></span><span class="bold"><strong>
        ;</strong></span>]<sup>+</sup> |</p><p><span class="bold"><strong> focal-set:</strong></span>
        [<span class="emphasis"><em>idf</em></span> <span class="emphasis"><em>inequality-term</em></span>
        <span class="emphasis"><em>boundary-parameter</em></span><span class="bold"><strong>
        ;</strong></span>]<sup>+</sup> |</p><p><span class="bold"><strong> derivative:</strong></span> [<span class="bold"><strong>d/dt </strong></span><span class="emphasis"><em>idf</em></span> ( <span class="bold"><strong>Undefined</strong></span> | <span class="bold"><strong>Zero</strong></span> | <span class="bold"><strong>Positive</strong></span> | <span class="bold"><strong>Negative</strong></span> | <span class="bold"><strong>All</strong></span>
        ) <span class="bold"><strong>;</strong></span>]<sup>+</sup>
        |</p><p><span class="bold"><strong> state:</strong></span> [( <span class="bold"><strong>IsSteadyState;</strong></span> | <span class="bold"><strong>IsInitialState;</strong></span> | <span class="bold"><strong>IsInSCC;</strong></span> | <span class="bold"><strong>IsInTerminalCycle;</strong></span> )
        ]<sup>+</sup></p><p><span class="emphasis"><em>inequality-term</em></span> ::= <span class="bold"><strong>&gt;</strong></span> | <span class="bold"><strong>&gt;=</strong></span> |
        <span class="bold"><strong>&lt;</strong></span> | <span class="bold"><strong>&lt;=</strong></span></p><p><span class="emphasis"><em>property-formula</em></span> ::=
        [^;]<sup>*</sup></p><p><span class="emphasis"><em>boundary-parameter</em></span> ::=
        <span class="emphasis"><em>idf</em></span> |
        <span class="emphasis"><em>focal-param-expression</em></span></p><p><span class="emphasis"><em>focal-param-expression ::= synthesis-param-sum
        <span class="bold"><strong>/</strong></span>
        degradation-param-sum</em></span></p><p><span class="emphasis"><em>synthesis-param-sum ::= idf | <span class="bold"><strong>(</strong></span> idf [<span class="bold"><strong>+</strong></span>
        idf]<sup>+</sup> <span class="bold"><strong>)</strong></span>
        </em></span></p><p><span class="emphasis"><em>degradation-param-sum ::= idf | <span class="bold"><strong>(</strong></span> idf [<span class="bold"><strong>+</strong></span>
        idf]<sup>+</sup> <span class="bold"><strong>)</strong></span></em></span></p><p><span class="emphasis"><em>idf</em></span> ::= (<span class="bold"><strong>a</strong></span>
        |..| <span class="bold"><strong>z </strong></span>|<span class="bold"><strong>
        A</strong></span> |..| <span class="bold"><strong>Z</strong></span>) [<span class="bold"><strong>a</strong></span> |..| <span class="bold"><strong>z
        </strong></span>|<span class="bold"><strong> A</strong></span> |..| <span class="bold"><strong>Z </strong></span>|<span class="bold"><strong> 0</strong></span> |..|
        <span class="bold"><strong>9 </strong></span>|<span class="bold"><strong>
        _</strong></span> ]<sup>*</sup></p></div><p>Comments can be inserted into the files. The syntax of comments
      follows the rules for comments in languages like C: multi-line comments
      are placed between <span class="bold"><strong>/*</strong></span> and <span class="bold"><strong>*/</strong></span>, while single-line comments are placed after
      characters <span class="bold"><strong>//</strong></span>.</p></div><div class="section" title="3.3.3.&nbsp;Additional constraints"><div class="titlepage"><div><div><h3 class="title"><a name="N11208"></a>3.3.3.&nbsp;Additional constraints</h3></div></div></div><p>In addition to the above syntax, a number of constraints must be
      respected:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem" style="list-style-type: disc"><p>The property formula is not checked when loading the property
          and the corresponding atomic propositions. For the grammar, the
          formula consists of everything until the semi colon. The syntax
          should be verified when trying to check the validity of the
          property.</p></li><li class="listitem" style="list-style-type: disc"><p>Atomic-propositions used inside a property formula must be
          defined. Their identifier is be the identifier appearing on the
          project tree.</p></li><li class="listitem" style="list-style-type: disc"><p>For the subatomic propositions domain and focal-set, the
          identifier <span class="emphasis"><em>idf</em></span> represents a variable name that
          must be declared in the model. For each variable, the lower and
          upper bounds should be defined. If absent the lower bound will be
          <span class="emphasis"><em>zero_var</em></span> and the upper bound
          <span class="emphasis"><em>max_var</em></span>.</p></li><li class="listitem" style="list-style-type: disc"><p>For the subatomic proposition derivative, the identifier
          <span class="emphasis"><em>idf</em></span> represents a variable name that must be
          declared in the model.</p></li><li class="listitem" style="list-style-type: disc"><p>The property block must appear exactly once in a variable
          block.</p></li></ul></div></div><div class="section" title="3.3.4.&nbsp;BNF of a property formula"><div class="titlepage"><div><div><h3 class="title"><a name="N1122B"></a>3.3.4.&nbsp;BNF of a property formula</h3></div></div></div><p>The BNF syntax of a property formula is given below. The axiom is
      <span class="emphasis"><em>specification</em></span>.</p><p>Some of these operators are boolean operators (<span class="bold"><strong>nil</strong></span>, <span class="bold"><strong>not</strong></span>,
      <span class="bold"><strong>true</strong></span>, <span class="bold"><strong>false</strong></span>, <span class="bold"><strong>and</strong></span>,
      <span class="bold"><strong>or</strong></span>, <span class="bold"><strong>equ</strong></span>, <span class="bold"><strong>implies</strong></span>),
      CTL and CTRL operators (<span class="bold"><strong>EF</strong></span>, <span class="bold"><strong>AF</strong></span>, <span class="bold"><strong>EX</strong></span>,
      <span class="bold"><strong>AX</strong></span>, <span class="bold"><strong>EG</strong></span>, <span class="bold"><strong>AG</strong></span>,
      <span class="bold"><strong>A</strong></span>, <span class="bold"><strong>E</strong></span>,
      <span class="bold"><strong>U</strong></span>, <span class="bold"><strong>R</strong></span>,
      <span class="bold"><strong>EF@</strong></span>, <span class="bold"><strong>AF@</strong></span>, <span class="bold"><strong>EG-|</strong></span>,
      <span class="bold"><strong>AG-|</strong></span>) and regular operators (<span class="bold"><strong>.</strong></span>, <span class="bold"><strong>|</strong></span>, <span class="bold"><strong>*</strong></span>,<span class="bold"><strong>+</strong></span>). The CTRL
      grammar describing how these operators are articulated is shown
      below.</p><div class="sidebar"><p class="title"></p><p><span class="emphasis"><em>specification</em></span> ::=
        <span class="emphasis"><em>equ-state-formula </em></span><span class="HTML Code">;</span></p><p><span class="emphasis"><em>equ-state-formula</em></span> ::=
        <span class="emphasis"><em>equ-state-formula </em></span><span class="bold"><strong>&lt;equ&gt;</strong></span><span class="emphasis"><em> implies-state-formula
        </em></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="emphasis"><em>implies-state-formula
        totototo</em></span>;</p><p><span class="emphasis"><em>implies-state-formula</em></span> ::=
        <span class="emphasis"><em>implies-state-formula </em></span><span class="bold"><strong>&lt;implies&gt;</strong></span><span class="emphasis"><em>
        or-state-formula</em></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="emphasis"><em>or-state-formula
        </em></span>;</p><p><span class="emphasis"><em>or-state-formula</em></span> ::=
        <span class="emphasis"><em>or-state-formula </em></span><span class="bold"><strong>&lt;or&gt;</strong></span><span class="emphasis"><em>
        and-state-formula</em></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="emphasis"><em>and-state-formula
        </em></span>;</p><p><span class="emphasis"><em>and-state-formula</em></span> ::=
        <span class="emphasis"><em>and-state-formula </em></span><span class="bold"><strong>&lt;and&gt;</strong></span><span class="emphasis"><em>
        unary-state-formula</em></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="emphasis"><em>unary-state-formula
        </em></span>;</p><p><span class="emphasis"><em>unary-state-formula</em></span> ::= <span class="emphasis"><em><span class="bold"><strong>&lt;not&gt;</strong></span>
        unary-state-formula</em></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|
        <span class="emphasis"><em>temporal-operator</em></span><span class="emphasis"><em>
        unary-state-formula</em></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="emphasis"><em>basic-state-formula
        </em></span>;</p><p><span class="emphasis"><em>temporal-operator</em></span> ::= <span class="bold"><strong>&lt;EF&gt; &lt;{&gt;</strong></span><span class="emphasis"><em> regular-formula
        </em></span><span class="bold"><strong>&lt;}&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;AF&gt;
        &lt;{&gt;</strong></span><span class="emphasis"><em> regular-formula </em></span><span class="bold"><strong>&lt;}&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| &lt;<span class="bold"><strong><span class="bold"><strong>EG&gt;
        &lt;{&gt;</strong></span></strong></span><span class="emphasis"><em> regular-formula
        </em></span><span class="bold"><strong>&lt;}&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;AG&gt;
        &lt;{&gt;</strong></span><span class="emphasis"><em> regular-formula </em></span><span class="bold"><strong>&lt;}&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="emphasis"><em>ctl-unary-operator
        </em></span>;</p><p><span class="emphasis"><em>ctl-unary-operator</em></span> ::= <span class="bold"><strong>&lt;EX&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;AX&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;EF&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;AF&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;EG&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;AG&gt;</strong></span>;</p><p><span class="emphasis"><em>basic-state-formula</em></span> ::= <span class="bold"><strong>&lt;EF&gt; &lt;{&gt;</strong></span><span class="emphasis"><em> regular-formula
        </em></span><span class="bold"><strong>&lt;}&gt; &lt;@&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;AF&gt; &lt;{&gt;</strong></span><span class="emphasis"><em> regular-formula
        </em></span><span class="bold"><strong>&lt;}&gt; &lt;@&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;EG&gt; &lt;{&gt;</strong></span><span class="emphasis"><em> regular-formula
        </em></span><span class="bold"><strong>&lt;}&gt;
        &lt;-|&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;AG&gt; &lt;{&gt;</strong></span><span class="emphasis"><em> regular-formula
        </em></span><span class="bold"><strong>&lt;}&gt;
        &lt;-|&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|
        <span class="emphasis"><em>atomic-formula</em></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;(&gt; state-formula </strong></span><span class="bold"><strong>&lt;)&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="emphasis"><em>ctl-binary-formula
        </em></span>;</p><p><span class="emphasis"><em>ctl-binary-formula</em></span> ::= <span class="bold"><strong>&lt;E&gt; &lt;[&gt;</strong></span><span class="emphasis"><em> state-formula
        </em></span><span class="bold"><strong>&lt;U&gt;</strong></span><span class="emphasis"><em>
        state-formula </em></span><span class="bold"><strong>&lt;]&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;A&gt;
        &lt;[&gt;</strong></span><span class="emphasis"><em> state-formula </em></span><span class="bold"><strong>&lt;U&gt;</strong></span><span class="emphasis"><em> state-formula
        </em></span><span class="bold"><strong>&lt;]&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;E&gt;
        &lt;[&gt;</strong></span><span class="emphasis"><em> state-formula </em></span><span class="bold"><strong>&lt;R&gt;</strong></span><span class="emphasis"><em> state-formula</em></span>
        <span class="bold"><strong>&lt;]&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;A&gt;
        &lt;[&gt;</strong></span><span class="emphasis"><em> state-formula </em></span><span class="bold"><strong>&lt;R&gt;"</strong></span><span class="emphasis"><em> state-formula
        </em></span><span class="bold"><strong>&lt;]&gt;</strong></span> ;</p><p><span class="emphasis"><em>atomic-formula</em></span> ::= <span class="bold"><strong>&lt;TRUE&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;FALSE&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt; "
        &gt;</strong></span>[<span class="emphasis"><em>tmp</em></span>
        ]<sup>*</sup><span class="bold"><strong>&lt; "
        &gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt; '
        &gt;</strong></span>[ <span class="emphasis"><em>tmp</em></span>
        ]<sup>*</sup><span class="bold"><strong>&lt; '
        &gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="emphasis"><em>identifier
        </em></span>;</p><p><span class="emphasis"><em>regular-formula</em></span> ::=
        <span class="emphasis"><em>concatenation-regular-formula </em></span>;</p><p><span class="emphasis"><em>concatenation-regular-formula</em></span> ::=
        <span class="emphasis"><em>concatenation-regular-formula </em></span><span class="bold"><strong>&lt;.&gt;</strong></span><span class="emphasis"><em>
        choice-regular-formula</em></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|
        <span class="emphasis"><em>choice-regular-formula </em></span>;</p><p><span class="emphasis"><em>choice-regular-formula</em></span> ::=
        <span class="emphasis"><em>choice-regular-formula </em></span><span class="bold"><strong>&lt;|&gt;</strong></span><span class="emphasis"><em>
        unary-regular-formula</em></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|
        <span class="emphasis"><em>unary-regular-formula </em></span>;</p><p><span class="emphasis"><em>unary-regular-formula</em></span> ::=
        <span class="emphasis"><em>unary-regular-formula </em></span><span class="bold"><strong>&lt;*&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|
        <span class="emphasis"><em>unary-regular-formula </em></span><span class="bold"><strong>&lt;+&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|
        <span class="emphasis"><em>basic-regular-formula</em></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="emphasis"><em>state-formula
        </em></span>;</p><p><span class="emphasis"><em>basic-regular-formula</em></span> ::= <span class="bold"><strong>&lt;NIL&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;(&gt;</strong></span><span class="emphasis"><em>
        concatenation-regular-formula </em></span><span class="bold"><strong>&lt;.&gt;</strong></span><span class="emphasis"><em>
        choice-regular-formula</em></span> <span class="bold"><strong>&lt;)&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;(</strong></span><span class="emphasis"><em>&gt; choice-regular-formula
        </em></span><span class="bold"><strong>&lt;|&gt;</strong></span><span class="emphasis"><em>
        unary-regular-formula </em></span><span class="bold"><strong>&lt;)&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;(&gt;</strong></span><span class="emphasis"><em> unary-regular-formula
        </em></span><span class="bold"><strong>&lt;*&gt; </strong></span><span class="bold"><strong>&lt;)&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;(&gt;</strong></span><span class="emphasis"><em> unary-regular-formula
        </em></span><span class="bold"><strong>&lt;+&gt; </strong></span><span class="bold"><strong>&lt;)&gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt;(&gt;</strong></span><span class="emphasis"><em> basic-regular-formula
        </em></span><span class="bold"><strong>&lt;)&gt;</strong></span>;</p><p><span class="emphasis"><em>tmp</em></span> ::= [<span class="bold"><strong>A</strong></span>-<span class="bold"><strong>Z_0</strong></span>-<span class="bold"><strong>9a</strong></span>-<span class="bold"><strong>z</strong></span> ]</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt; \" &gt;</strong></span></p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| <span class="bold"><strong>&lt; \' &gt;</strong></span>
        ;</p><p><span class="emphasis"><em>identifier</em></span> ::= (<span class="bold"><strong>&lt;a&gt;</strong></span> |..| <span class="bold"><strong>&lt;z&gt;
        </strong></span>|<span class="bold"><strong> &lt;A&gt;</strong></span> |..| <span class="bold"><strong>&lt;Z&gt;</strong></span>) [<span class="bold"><strong>&lt;a&gt;</strong></span> |..| <span class="bold"><strong>&lt;z&gt;
        </strong></span>|<span class="bold"><strong> &lt;A&gt;</strong></span> |..| <span class="bold"><strong>&lt;Z&gt; </strong></span>|<span class="bold"><strong>
        &lt;0&gt;</strong></span> |..| <span class="bold"><strong>&lt;9&gt;
        </strong></span>|<span class="bold"><strong> &lt;_&gt;</strong></span>
        ]<sup>*</sup> ;</p></div></div></div><div class="section" title="3.4.&nbsp;CUP PARSER GENERATOR COPYRIGHT NOTICE, LICENSE AND DISCLAIMER"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="N114D2"></a>3.4.&nbsp;CUP PARSER GENERATOR COPYRIGHT NOTICE, LICENSE AND
    DISCLAIMER</h2></div></div></div><div class="sidebar"><p class="title"></p><p>Copyright 1996 by Scott Hudson, Frank Flannery, C. Scott
      Ananian</p><p>Permission to use, copy, modify, and distribute this software and
      its documentation for any purpose and without fee is hereby granted,
      provided that the above copyright notice appear in all copies and that
      both the copyright notice and this permission notice and warranty
      disclaimer appear in supporting documentation, and that the names of the
      authors or their employers not be used in advertising or publicity
      pertaining to distribution of the software without specific, written
      prior permission.</p><p>The authors and their employers disclaim all warranties with
      regard to this software, including all implied warranties of
      merchantability and fitness. In no event shall the authors or their
      employers be liable for any special, indirect or consequential damages
      or any damages whatsoever resulting from loss of use, data or profits,
      whether in an action of contract, negligence or other tortious action,
      arising out of or in connection with the use or performance of this
      software.</p></div></div><div class="section" title="3.5.&nbsp;JGRAPH COPYRIGHT NOTICE, LICENSE, AND DISCLAIMER"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="N114DE"></a>3.5.&nbsp;JGRAPH COPYRIGHT NOTICE, LICENSE, AND DISCLAIMER</h2></div></div></div><div class="sidebar"><p class="title"></p><p>Copyright (c) 2001-2004, Gaudenz</p><p>Alder All rights reserved.</p><p>Redistribution and use in source and binary forms, with or without
      modification, are permitted provided that the following conditions are
      met:</p><div class="itemizedlist"><ul class="itemizedlist" type="disc"><li class="listitem"><p>Redistributions of source code must retain the above
            copyright notice, this list of conditions and the following
            disclaimer.</p></li><li class="listitem"><p>Redistributions in binary form must reproduce the above
            copyright notice, this list of conditions and the following
            disclaimer in the documentation and/or other materials provided
            with the distribution.</p></li><li class="listitem"><p>Neither the name of JGraph nor the names of its contributors
            may be used to endorse or promote products derived from this
            software without specific prior written permission.</p></li></ul></div><p>THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
      CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,
      BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
      FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
      COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
      INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
      NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
      USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
      ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
      (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
      THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.</p></div></div><div class="section" title="3.6.&nbsp;SAT4J LICENSE"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="N114F6"></a>3.6.&nbsp;SAT4J LICENSE</h2></div></div></div><div class="sidebar"><p class="title"></p><p>SAT4J: a SATisfiability library for Java</p><p>Copyright (C) 2004 Daniel Le Berre</p><p>Based on the original minisat specification from:</p><p>An extensible SAT solver. Niklas E&eacute;n and Niklas S&ouml;rensson.
      Proceedings of the Sixth International Conference on Theory and
      Applications of Satisfiability Testing, LNCS 2919, pp 502-518,
      2003.</p><p>This library is free software; you can redistribute it and/or
      modify it under the terms of the GNU Lesser General Public License as
      published by the Free Software Foundation; either version 2.1 of the
      License, or (at your option) any later version.</p><p>This library is distributed in the hope that it will be useful,
      but WITHOUT ANY WARRANTY; without even the implied warranty of
      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
      General Public License for more details.</p><p>You should have received a copy of the GNU Lesser General Public
      License along with this library; if not, write to the Free Software
      Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
      USA</p></div></div><div class="section" title="3.7.&nbsp;NuSMV LICENSE"><div class="titlepage"><div><div><h2 class="title" style="clear: both"><a name="N1150A"></a>3.7.&nbsp;NuSMV LICENSE</h2></div></div></div><div class="sidebar"><p class="title"></p><p>NuSMV version 2 (NuSMV 2 in short) is licensed under the GNU
      Lesser General Public License (LGPL in short). File LGLP-2.1 contains a
      copy of the License.</p><p>The aim of the NuSMV OpenSource project is to allow the whole
      model checking community to participate to the development of NuSMV. To
      this purpose, we have chosen a license that:</p><div class="orderedlist"><ol class="orderedlist" type="1"><li class="listitem"><p>is "copyleft", that is, it requires that anyone who improves
            the system has to make the improvements freely available;</p></li><li class="listitem"><p>permits to use the system in research and commercial
            applications, without restrictions.</p></li></ol></div><p>In brief, the LGPL license allows anyone to freely download, copy,
      use, modify, and redistribute NuSMV 2, proviso that any modification
      and/or extension to the library is made publicly available under the
      terms of LGPL.</p><p>The license also allows the usage of the NuSMV 2 as part of a
      larger software system <span class="bold"><strong>without</strong></span> being
      obliged to distributing the whole software under LGPL. Also in this
      case, the modification to NuSMV 2 (<span class="bold"><strong>not</strong></span>
      to the larger software) should be made available under LGPL.</p><p>The precise terms and conditions for copying, distribution and
      modification can be found in file LGPL-2.1. You can contact
      <code class="email">&lt;<a class="email" href="mailto:nusmv-users@irst.itc.it">nusmv-users@irst.itc.it</a>&gt;</code> if you have any doubt or comment
      on the license.</p></div></div></div></div></body></html>